From 025d4aae3badd0a04ab562ae65df616fe306d145 Mon Sep 17 00:00:00 2001
From: Joe Schubert <joseph.schubert@gmail.com>
Date: Wed, 23 Aug 2017 23:54:13 +0200
Subject: [PATCH] Update-binary: stop calling log commands that do not exist in TWRP

diff --git a/updater/blockimg.cpp b/updater/blockimg.cpp
index a0b9ad2..5f17283 100644
--- a/updater/blockimg.cpp
+++ b/updater/blockimg.cpp
@@ -1599,11 +1599,6 @@ static Value* PerformBlockImageUpdate(const char* name, State* state,
     LOG(INFO) << "max alloc needed was " << params.buffer.size();
 
     const char* partition = strrchr(blockdev_filename->data.c_str(), '/');
-    if (partition != nullptr && *(partition + 1) != 0) {
-      fprintf(cmd_pipe, "log bytes_written_%s: %zu\n", partition + 1, params.written * BLOCKSIZE);
-      fprintf(cmd_pipe, "log bytes_stashed_%s: %zu\n", partition + 1, params.stashed * BLOCKSIZE);
-      fflush(cmd_pipe);
-    }
     // Delete stash only after successfully completing the update, as it may contain blocks needed
     // to complete the update later.
     DeleteStash(params.stashbase);
