From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Allow twrp parted binary

 * BoardConfig flag: 'TW_EXCLUDE_PARTED := true'

diff --git a/Android.mk b/Android.mk
index 85cdd32..926010e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -429,6 +429,9 @@
         LOCAL_ADDITIONAL_DEPENDENCIES += sgdisk_static
     endif
 endif
+ifeq ($(TW_EXCLUDE_PARTED), true)
+    LOCAL_ADDITIONAL_DEPENDENCIES += parted
+endif
 ifneq ($(TW_EXCLUDE_ENCRYPTED_BACKUPS), true)
     LOCAL_ADDITIONAL_DEPENDENCIES += openaes openaes_license
 endif
diff --git a/prebuilt/Android.mk b/prebuilt/Android.mk
index f88f48f..0604909 100644
--- a/prebuilt/Android.mk
+++ b/prebuilt/Android.mk
@@ -354,16 +354,19 @@
 	    LOCAL_SRC_FILES := $(LOCAL_MODULE)
 	    include $(BUILD_PREBUILT)
 	endif
-	#parted
-	#include $(CLEAR_VARS)
-	#LOCAL_MODULE := parted
-	#LOCAL_MODULE_TAGS := eng
-	#LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
-	#LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
-	#LOCAL_SRC_FILES := $(LOCAL_MODULE)
-	#include $(BUILD_PREBUILT)
 endif
 
+ifeq ($(TW_EXCLUDE_PARTED), true)
+	#copy parted binary
+	include $(CLEAR_VARS)
+	LOCAL_MODULE := parted
+	LOCAL_MODULE_TAGS := eng
+	LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
+	LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
+	LOCAL_SRC_FILES := $(LOCAL_MODULE)
+	include $(BUILD_PREBUILT)
+endif
+
 # copy license file for OpenAES
 ifneq ($(TW_EXCLUDE_ENCRYPTED_BACKUPS), true)
 	include $(CLEAR_VARS)
