diff --git a/Android.mk b/Android.mk
index 10f6431..f725746 100644
--- a/Android.mk
+++ b/Android.mk
@@ -78,7 +78,7 @@ RECOVERY_API_VERSION := 3
 RECOVERY_FSTAB_VERSION := 2
 LOCAL_CFLAGS += -DRECOVERY_API_VERSION=$(RECOVERY_API_VERSION)
 LOCAL_CFLAGS += -Wno-unused-parameter
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 #LOCAL_STATIC_LIBRARIES := \
 #    libext4_utils_static \
@@ -547,7 +547,7 @@ endif
 # ===============================
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := fuse_sideload.cpp
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_CFLAGS := -O2 -g -DADB_HOST=0 -Wall -Wno-unused-parameter
 LOCAL_CFLAGS += -D_XOPEN_SOURCE -D_GNU_SOURCE
 
@@ -600,7 +600,7 @@ include $(BUILD_SHARED_LIBRARY)
 
 # All the APIs for testing
 include $(CLEAR_VARS)
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_MODULE := libverifier
 LOCAL_MODULE_TAGS := tests
 LOCAL_SRC_FILES := \
diff --git a/applypatch/Android.mk b/applypatch/Android.mk
index 581e53a..94f0ec4 100644
--- a/applypatch/Android.mk
+++ b/applypatch/Android.mk
@@ -29,7 +29,7 @@ LOCAL_C_INCLUDES += \
     external/zlib \
     $(commands_recovery_local_path)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := applypatch.cpp bspatch.cpp freecache.cpp imgpatch.cpp utils.cpp
 LOCAL_MODULE := libapplypatch
 LOCAL_MODULE_TAGS := eng
@@ -43,7 +43,7 @@ include $(CLEAR_VARS)
 ifeq ($(HOST_OS),linux)
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := bspatch.cpp imgpatch.cpp utils.cpp
 LOCAL_MODULE := libimgpatch
 LOCAL_C_INCLUDES += bootable/recovery
@@ -55,7 +55,7 @@ endif  # HOST_OS == linux
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := main.cpp
 LOCAL_MODULE := applypatch
 LOCAL_C_INCLUDES += bootable/recovery
@@ -68,7 +68,7 @@ include $(BUILD_EXECUTABLE)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := imgdiff.cpp utils.cpp bsdiff.cpp
 LOCAL_MODULE := imgdiff
 LOCAL_FORCE_STATIC_EXECUTABLE := true
diff --git a/edify/Android.mk b/edify/Android.mk
index 71cf765..dac7153 100644
--- a/edify/Android.mk
+++ b/edify/Android.mk
@@ -21,7 +21,7 @@ LOCAL_MODULE := edify
 LOCAL_YACCFLAGS := -v
 LOCAL_CPPFLAGS += -Wno-unused-parameter
 LOCAL_CPPFLAGS += -Wno-deprecated-register
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/..
 LOCAL_STATIC_LIBRARIES += libbase
 
@@ -37,7 +37,7 @@ LOCAL_SRC_FILES := $(edify_src_files)
 LOCAL_CPPFLAGS := -Wno-unused-parameter
 LOCAL_CPPFLAGS += -Wno-deprecated-register
 LOCAL_MODULE := libedify
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/..
 LOCAL_STATIC_LIBRARIES += libbase
 
diff --git a/minadbd/Android.mk b/minadbd/Android.mk
index 0427912..52544ec 100644
--- a/minadbd/Android.mk
+++ b/minadbd/Android.mk
@@ -3,7 +3,7 @@
 LOCAL_PATH:= $(call my-dir)
 
 minadbd_cflags := \
-    -Wall -Werror \
+    -Wall  \
     -Wno-unused-parameter \
     -Wno-missing-field-initializers \
     -DADB_HOST=0 \
@@ -17,7 +17,7 @@ LOCAL_SRC_FILES := \
     services.cpp \
     ../fuse_sideload.cpp
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_MODULE := libminadbd
 LOCAL_CFLAGS := $(minadbd_cflags)
 LOCAL_CONLY_FLAGS := -Wimplicit-function-declaration
diff --git a/minzip/Android.mk b/minzip/Android.mk
index 957ab0b..535aa3f 100644
--- a/minzip/Android.mk
+++ b/minzip/Android.mk
@@ -24,9 +24,9 @@ LOCAL_MODULE := libminzip
 
 LOCAL_SHARED_LIBRARIES += libz
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
-LOCAL_CFLAGS += -Werror -Wall
+LOCAL_CFLAGS +=  -Wall
 
 include $(BUILD_SHARED_LIBRARY)
 
@@ -56,8 +56,8 @@ LOCAL_MODULE := libminzip
 
 LOCAL_STATIC_LIBRARIES += libz
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
-LOCAL_CFLAGS += -Werror -Wall
+LOCAL_CFLAGS +=  -Wall
 
 include $(BUILD_STATIC_LIBRARY)
diff --git a/mtdutils/Android.mk b/mtdutils/Android.mk
index 7e5fadc..e4148bd 100644
--- a/mtdutils/Android.mk
+++ b/mtdutils/Android.mk
@@ -14,7 +14,7 @@ endif
 
 LOCAL_MODULE := libmtdutils
 LOCAL_STATIC_LIBRARIES := libcutils libc
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 include $(BUILD_STATIC_LIBRARY)
 
@@ -31,7 +31,7 @@ endif
 
 LOCAL_MODULE := libmtdutils
 LOCAL_SHARED_LIBRARIES := libcutils libc
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 include $(BUILD_SHARED_LIBRARY)
 
diff --git a/otafault/Android.mk b/otafault/Android.mk
index ba7add8..dad5b0c 100644
--- a/otafault/Android.mk
+++ b/otafault/Android.mk
@@ -25,7 +25,7 @@ otafault_static_libs := \
 LOCAL_SRC_FILES := config.cpp ota_io.cpp
 LOCAL_MODULE_TAGS := eng
 LOCAL_MODULE := libotafault
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES := bootable/recovery
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
 LOCAL_WHOLE_STATIC_LIBRARIES := $(otafault_static_libs)
diff --git a/toolbox/Android.mk b/toolbox/Android.mk
index 4c3e433..366137d 100644
--- a/toolbox/Android.mk
+++ b/toolbox/Android.mk
@@ -183,7 +183,7 @@ endif
 ifneq (,$(filter $(PLATFORM_SDK_VERSION), 21 22 23))
     LOCAL_CFLAGS += \
         -std=gnu99 \
-        -Werror -Wno-unused-parameter \
+         -Wno-unused-parameter \
         -I$(LOCAL_PATH)/upstream-netbsd/include \
         -include bsd-compatibility.h
 endif
diff --git a/uncrypt/Android.mk b/uncrypt/Android.mk
index debec7f..8a92025 100644
--- a/uncrypt/Android.mk
+++ b/uncrypt/Android.mk
@@ -16,7 +16,7 @@ LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 LOCAL_C_INCLUDES += $(commands_recovery_local_path)
 
diff --git a/update_verifier/Android.mk b/update_verifier/Android.mk
index 2bfd016..6a7cf86 100644
--- a/update_verifier/Android.mk
+++ b/update_verifier/Android.mk
@@ -16,7 +16,7 @@ LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := update_verifier.cpp
 
 LOCAL_MODULE := update_verifier
diff --git a/updater/Android.mk b/updater/Android.mk
index ea31a8b..b39800f 100644
--- a/updater/Android.mk
+++ b/updater/Android.mk
@@ -29,7 +29,7 @@ include $(CLEAR_VARS)
 # needed only for OTA packages.)
 LOCAL_MODULE_TAGS := eng
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 LOCAL_SRC_FILES := $(updater_src_files)
 
diff --git a/Android.mk b/Android.mk
index 10f6431..f725746 100644
--- a/Android.mk
+++ b/Android.mk
@@ -78,7 +78,7 @@ RECOVERY_API_VERSION := 3
 RECOVERY_FSTAB_VERSION := 2
 LOCAL_CFLAGS += -DRECOVERY_API_VERSION=$(RECOVERY_API_VERSION)
 LOCAL_CFLAGS += -Wno-unused-parameter
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 #LOCAL_STATIC_LIBRARIES := \
 #    libext4_utils_static \
@@ -547,7 +547,7 @@ endif
 # ===============================
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := fuse_sideload.cpp
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_CFLAGS := -O2 -g -DADB_HOST=0 -Wall -Wno-unused-parameter
 LOCAL_CFLAGS += -D_XOPEN_SOURCE -D_GNU_SOURCE
 
@@ -600,7 +600,7 @@ include $(BUILD_SHARED_LIBRARY)
 
 # All the APIs for testing
 include $(CLEAR_VARS)
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_MODULE := libverifier
 LOCAL_MODULE_TAGS := tests
 LOCAL_SRC_FILES := \
diff --git a/applypatch/Android.mk b/applypatch/Android.mk
index 581e53a..94f0ec4 100644
--- a/applypatch/Android.mk
+++ b/applypatch/Android.mk
@@ -29,7 +29,7 @@ LOCAL_C_INCLUDES += \
     external/zlib \
     $(commands_recovery_local_path)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := applypatch.cpp bspatch.cpp freecache.cpp imgpatch.cpp utils.cpp
 LOCAL_MODULE := libapplypatch
 LOCAL_MODULE_TAGS := eng
@@ -43,7 +43,7 @@ include $(CLEAR_VARS)
 ifeq ($(HOST_OS),linux)
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := bspatch.cpp imgpatch.cpp utils.cpp
 LOCAL_MODULE := libimgpatch
 LOCAL_C_INCLUDES += bootable/recovery
@@ -55,7 +55,7 @@ endif  # HOST_OS == linux
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := main.cpp
 LOCAL_MODULE := applypatch
 LOCAL_C_INCLUDES += bootable/recovery
@@ -68,7 +68,7 @@ include $(BUILD_EXECUTABLE)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := imgdiff.cpp utils.cpp bsdiff.cpp
 LOCAL_MODULE := imgdiff
 LOCAL_FORCE_STATIC_EXECUTABLE := true
diff --git a/edify/Android.mk b/edify/Android.mk
index 71cf765..dac7153 100644
--- a/edify/Android.mk
+++ b/edify/Android.mk
@@ -21,7 +21,7 @@ LOCAL_MODULE := edify
 LOCAL_YACCFLAGS := -v
 LOCAL_CPPFLAGS += -Wno-unused-parameter
 LOCAL_CPPFLAGS += -Wno-deprecated-register
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/..
 LOCAL_STATIC_LIBRARIES += libbase
 
@@ -37,7 +37,7 @@ LOCAL_SRC_FILES := $(edify_src_files)
 LOCAL_CPPFLAGS := -Wno-unused-parameter
 LOCAL_CPPFLAGS += -Wno-deprecated-register
 LOCAL_MODULE := libedify
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/..
 LOCAL_STATIC_LIBRARIES += libbase
 
diff --git a/minadbd/Android.mk b/minadbd/Android.mk
index 0427912..52544ec 100644
--- a/minadbd/Android.mk
+++ b/minadbd/Android.mk
@@ -3,7 +3,7 @@
 LOCAL_PATH:= $(call my-dir)
 
 minadbd_cflags := \
-    -Wall -Werror \
+    -Wall  \
     -Wno-unused-parameter \
     -Wno-missing-field-initializers \
     -DADB_HOST=0 \
@@ -17,7 +17,7 @@ LOCAL_SRC_FILES := \
     services.cpp \
     ../fuse_sideload.cpp
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_MODULE := libminadbd
 LOCAL_CFLAGS := $(minadbd_cflags)
 LOCAL_CONLY_FLAGS := -Wimplicit-function-declaration
diff --git a/minzip/Android.mk b/minzip/Android.mk
index 957ab0b..535aa3f 100644
--- a/minzip/Android.mk
+++ b/minzip/Android.mk
@@ -24,9 +24,9 @@ LOCAL_MODULE := libminzip
 
 LOCAL_SHARED_LIBRARIES += libz
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
-LOCAL_CFLAGS += -Werror -Wall
+LOCAL_CFLAGS +=  -Wall
 
 include $(BUILD_SHARED_LIBRARY)
 
@@ -56,8 +56,8 @@ LOCAL_MODULE := libminzip
 
 LOCAL_STATIC_LIBRARIES += libz
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
-LOCAL_CFLAGS += -Werror -Wall
+LOCAL_CFLAGS +=  -Wall
 
 include $(BUILD_STATIC_LIBRARY)
diff --git a/mtdutils/Android.mk b/mtdutils/Android.mk
index 7e5fadc..e4148bd 100644
--- a/mtdutils/Android.mk
+++ b/mtdutils/Android.mk
@@ -14,7 +14,7 @@ endif
 
 LOCAL_MODULE := libmtdutils
 LOCAL_STATIC_LIBRARIES := libcutils libc
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 include $(BUILD_STATIC_LIBRARY)
 
@@ -31,7 +31,7 @@ endif
 
 LOCAL_MODULE := libmtdutils
 LOCAL_SHARED_LIBRARIES := libcutils libc
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 include $(BUILD_SHARED_LIBRARY)
 
diff --git a/otafault/Android.mk b/otafault/Android.mk
index ba7add8..dad5b0c 100644
--- a/otafault/Android.mk
+++ b/otafault/Android.mk
@@ -25,7 +25,7 @@ otafault_static_libs := \
 LOCAL_SRC_FILES := config.cpp ota_io.cpp
 LOCAL_MODULE_TAGS := eng
 LOCAL_MODULE := libotafault
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_C_INCLUDES := bootable/recovery
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
 LOCAL_WHOLE_STATIC_LIBRARIES := $(otafault_static_libs)
diff --git a/toolbox/Android.mk b/toolbox/Android.mk
index 4c3e433..366137d 100644
--- a/toolbox/Android.mk
+++ b/toolbox/Android.mk
@@ -183,7 +183,7 @@ endif
 ifneq (,$(filter $(PLATFORM_SDK_VERSION), 21 22 23))
     LOCAL_CFLAGS += \
         -std=gnu99 \
-        -Werror -Wno-unused-parameter \
+         -Wno-unused-parameter \
         -I$(LOCAL_PATH)/upstream-netbsd/include \
         -include bsd-compatibility.h
 endif
diff --git a/uncrypt/Android.mk b/uncrypt/Android.mk
index debec7f..8a92025 100644
--- a/uncrypt/Android.mk
+++ b/uncrypt/Android.mk
@@ -16,7 +16,7 @@ LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 LOCAL_C_INCLUDES += $(commands_recovery_local_path)
 
diff --git a/update_verifier/Android.mk b/update_verifier/Android.mk
index 2bfd016..6a7cf86 100644
--- a/update_verifier/Android.mk
+++ b/update_verifier/Android.mk
@@ -16,7 +16,7 @@ LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 LOCAL_SRC_FILES := update_verifier.cpp
 
 LOCAL_MODULE := update_verifier
diff --git a/updater/Android.mk b/updater/Android.mk
index ea31a8b..b39800f 100644
--- a/updater/Android.mk
+++ b/updater/Android.mk
@@ -29,7 +29,7 @@ include $(CLEAR_VARS)
 # needed only for OTA packages.)
 LOCAL_MODULE_TAGS := eng
 
-LOCAL_CLANG := true
+LOCAL_CLANG := false
 
 LOCAL_SRC_FILES := $(updater_src_files)
 
