diff --git a/build/Android.common_build.mk b/build/Android.common_build.mk
index 2294ddb..68bd5b3 100644
--- a/build/Android.common_build.mk
+++ b/build/Android.common_build.mk
@@ -221,7 +221,7 @@ art_cflags := \
   -std=gnu++11 \
   -ggdb3 \
   -Wall \
-  -Werror \
+   \
   -Wextra \
   -Wstrict-aliasing \
   -fstrict-aliasing \
diff --git a/dalvikvm/Android.mk b/dalvikvm/Android.mk
index 71e9a28..2768e5e 100644
--- a/dalvikvm/Android.mk
+++ b/dalvikvm/Android.mk
@@ -18,7 +18,7 @@ LOCAL_PATH := $(call my-dir)
 
 include art/build/Android.common.mk
 
-dalvikvm_cflags := -Wall -Werror -Wextra -std=gnu++11
+dalvikvm_cflags := -Wall  -Wextra -std=gnu++11
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := dalvikvm
diff --git a/dexdump/Android.mk b/dexdump/Android.mk
index ec2529e..ecca045 100755
--- a/dexdump/Android.mk
+++ b/dexdump/Android.mk
@@ -34,6 +34,7 @@ LOCAL_C_INCLUDES := $(dexdump_c_includes)
 LOCAL_CFLAGS += -Wall
 LOCAL_SHARED_LIBRARIES += $(dexdump_libraries)
 LOCAL_MODULE := dexdump2
+LOCAL_CLANG := true
 include $(BUILD_EXECUTABLE)
 endif # !SDK_ONLY
 
@@ -48,5 +49,6 @@ LOCAL_C_INCLUDES := $(dexdump_c_includes)
 LOCAL_CFLAGS += -Wall
 LOCAL_SHARED_LIBRARIES += $(dexdump_libraries)
 LOCAL_MODULE := dexdump2
+LOCAL_CLANG := true
 LOCAL_MULTILIB := $(ART_MULTILIB_OVERRIDE_host)
 include $(BUILD_HOST_EXECUTABLE)
diff --git a/runtime/Android.mk b/runtime/Android.mk
index ce73030..5e374ec 100644
--- a/runtime/Android.mk
+++ b/runtime/Android.mk
@@ -15,7 +15,7 @@
 #
 
 LOCAL_PATH := $(call my-dir)
-
+LOCAL_CLANG := true
 include art/build/Android.common_build.mk
 
 LIBART_COMMON_SRC_FILES := \
@@ -444,6 +444,7 @@ define build-runtime-library
   endif
 
   LOCAL_MODULE_TAGS := optional
+  LOCAL_CLANG := true
 
   ifeq ($$(art_static_or_shared),static)
     LOCAL_MODULE_CLASS := STATIC_LIBRARIES
diff --git a/runtime/gc/collector/mark_sweep.cc b/runtime/gc/collector/mark_sweep.cc
index 24cbf10..97a7692 100644
--- a/runtime/gc/collector/mark_sweep.cc
+++ b/runtime/gc/collector/mark_sweep.cc
@@ -645,7 +645,7 @@ class MarkSweep::DelayReferenceReferentVisitor {
   MarkSweep* const collector_;
 };
 
-template <bool kUseFinger = false>
+template <bool kUseFinger>
 class MarkSweep::MarkStackTask : public Task {
  public:
   MarkStackTask(ThreadPool* thread_pool,
diff --git a/build/Android.common_build.mk b/build/Android.common_build.mk
index 2294ddb..68bd5b3 100644
--- a/build/Android.common_build.mk
+++ b/build/Android.common_build.mk
@@ -221,7 +221,7 @@ art_cflags := \
   -std=gnu++11 \
   -ggdb3 \
   -Wall \
-  -Werror \
+   \
   -Wextra \
   -Wstrict-aliasing \
   -fstrict-aliasing \
diff --git a/dalvikvm/Android.mk b/dalvikvm/Android.mk
index 71e9a28..2768e5e 100644
--- a/dalvikvm/Android.mk
+++ b/dalvikvm/Android.mk
@@ -18,7 +18,7 @@ LOCAL_PATH := $(call my-dir)
 
 include art/build/Android.common.mk
 
-dalvikvm_cflags := -Wall -Werror -Wextra -std=gnu++11
+dalvikvm_cflags := -Wall  -Wextra -std=gnu++11
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := dalvikvm
diff --git a/dexdump/Android.mk b/dexdump/Android.mk
index ec2529e..ecca045 100755
--- a/dexdump/Android.mk
+++ b/dexdump/Android.mk
@@ -34,6 +34,7 @@ LOCAL_C_INCLUDES := $(dexdump_c_includes)
 LOCAL_CFLAGS += -Wall
 LOCAL_SHARED_LIBRARIES += $(dexdump_libraries)
 LOCAL_MODULE := dexdump2
+LOCAL_CLANG := true
 include $(BUILD_EXECUTABLE)
 endif # !SDK_ONLY
 
@@ -48,5 +49,6 @@ LOCAL_C_INCLUDES := $(dexdump_c_includes)
 LOCAL_CFLAGS += -Wall
 LOCAL_SHARED_LIBRARIES += $(dexdump_libraries)
 LOCAL_MODULE := dexdump2
+LOCAL_CLANG := true
 LOCAL_MULTILIB := $(ART_MULTILIB_OVERRIDE_host)
 include $(BUILD_HOST_EXECUTABLE)
diff --git a/runtime/Android.mk b/runtime/Android.mk
index ce73030..5e374ec 100644
--- a/runtime/Android.mk
+++ b/runtime/Android.mk
@@ -15,7 +15,7 @@
 #
 
 LOCAL_PATH := $(call my-dir)
-
+LOCAL_CLANG := true
 include art/build/Android.common_build.mk
 
 LIBART_COMMON_SRC_FILES := \
@@ -444,6 +444,7 @@ define build-runtime-library
   endif
 
   LOCAL_MODULE_TAGS := optional
+  LOCAL_CLANG := true
 
   ifeq ($$(art_static_or_shared),static)
     LOCAL_MODULE_CLASS := STATIC_LIBRARIES
diff --git a/runtime/gc/collector/mark_sweep.cc b/runtime/gc/collector/mark_sweep.cc
index 24cbf10..97a7692 100644
--- a/runtime/gc/collector/mark_sweep.cc
+++ b/runtime/gc/collector/mark_sweep.cc
@@ -645,7 +645,7 @@ class MarkSweep::DelayReferenceReferentVisitor {
   MarkSweep* const collector_;
 };
 
-template <bool kUseFinger = false>
+template <bool kUseFinger>
 class MarkSweep::MarkStackTask : public Task {
  public:
   MarkStackTask(ThreadPool* thread_pool,
