diff --git a/Android.mk b/Android.mk
index 33a43d8..298c90b 100644
--- a/Android.mk
+++ b/Android.mk
@@ -442,15 +442,7 @@ ifeq ($(TW_INCLUDE_DUMLOCK), true)
 endif
 ifneq ($(TW_EXCLUDE_SUPERSU), true)
     LOCAL_ADDITIONAL_DEPENDENCIES += \
-        install-recovery.sh 99SuperSUDaemon Superuser.apk
-    ifeq ($(TARGET_ARCH), arm)
-        LOCAL_ADDITIONAL_DEPENDENCIES += \
-            chattr.pie libsupol.so suarm supolicy
-    endif
-    ifeq ($(TARGET_ARCH), arm64)
-        LOCAL_ADDITIONAL_DEPENDENCIES += \
-            libsupol.soarm64 suarm64 supolicyarm64
-    endif
+        suarm install-recovery.sh 99SuperSUDaemon Superuser.apk
 endif
 ifeq ($(TW_INCLUDE_FB2PNG), true)
     LOCAL_ADDITIONAL_DEPENDENCIES += fb2png
