Subject: [PATCH] Toybox: minicrom is not in the lineage source.

diff --git a/toybox/Android.mk b/toybox/Android.mk
index 43a893d..1cf273d 100644
--- a/toybox/Android.mk
+++ b/toybox/Android.mk
@@ -240,6 +240,11 @@
     toys/pending/ftpget.c
 endif
 
+ifneq ($(wildcard toys/net/microcom.c),)
+LOCAL_SRC_FILES += \
+    toys/net/microcom.c
+endif
+
 # Account for master branch changes pulld into CM14.1
 ifneq ($(CM_BUILD),)
 LOCAL_SRC_FILES += \
@@ -268,7 +273,6 @@
     toys/android/sendevent.c \
     toys/android/start.c \
     toys/net/ifconfig.c \
-    toys/net/microcom.c \
     toys/net/netcat.c \
     toys/net/netstat.c \
     toys/net/rfkill.c \
