diff --git a/core/Makefile b/core/Makefile
index 1539900..b22731f 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1348,6 +1348,9 @@
 		$(hide) cp $(PRODUCT_OUT)/ramdisk-recovery.cpio $(PRODUCT_OUT)/install/bin/recovery.cpio
 		@echo ----- Compressed recovery gzip ------
 		$(hide) gzip $(PRODUCT_OUT)/install/bin/recovery.cpio
+		@echo ----- Generating Changelog ------
+		$(hide) ./vendor/samsung/tools/changelog
+		$(hide) mv $(PRODUCT_OUT)/Changelog.txt $(PRODUCT_OUT)/$(LINEAGE_VERSION)-$(TARGET_PRODUCT).txt
 
 $(recovery_ramdisk): $(MINIGZIP) \
 		$(recovery_uncompressed_ramdisk)
