Subject: [PATCH] build: add TARGET_SPECIFIC_HEADER_PATH to config

diff --git a/core/soong_config.mk b/core/soong_config.mk
index 1b012ff..b01fe99 100644
--- a/core/soong_config.mk
+++ b/core/soong_config.mk
@@ -88,7 +88,8 @@
 	echo ''; \
 	echo '    "BtConfigIncludeDir": "$(BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR)",'; \
 	echo ''; \
-	echo '    "DeviceKernelHeaders": $(call json_list,$(strip $(TARGET_PROJECT_SYSTEM_INCLUDES)))'; \
+	echo '    "DeviceKernelHeaders": $(call json_list,$(strip $(TARGET_PROJECT_SYSTEM_INCLUDES))),'; \
+	echo '    "TargetHeaderPath": "$(TARGET_SPECIFIC_HEADER_PATH)"'; \
 	echo '}') >> $(SOONG_VARIABLES_TMP); \
 	if ! cmp -s $(SOONG_VARIABLES_TMP) $(SOONG_VARIABLES); then \
 	  mv $(SOONG_VARIABLES_TMP) $(SOONG_VARIABLES); \
