Compressed recovery gzip

diff --git a/core/Makefile b/core/Makefile
index 1539900..0227310 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1346,6 +1346,8 @@
 		$(hide) $(MKBOOTFS) $(TARGET_RECOVERY_ROOT_OUT) > $@
 		@echo ----- Copy uncompressed recovery ------
 		$(hide) cp $(PRODUCT_OUT)/ramdisk-recovery.cpio $(PRODUCT_OUT)/install/bin/recovery.cpio
+		@echo ----- Compressed recovery gzip ------
+		$(hide) gzip $(PRODUCT_OUT)/install/bin/recovery.cpio
 
 $(recovery_ramdisk): $(MINIGZIP) \
 		$(recovery_uncompressed_ramdisk)
