Subject:  soong: add STE compiler flags

diff --git a/android/makevars.go b/android/makevars.go
index 482fbde..000cc1c 100644
--- a/android/makevars.go
+++ b/android/makevars.go
@@ -194,7 +194,7 @@ my_check_failed := false
 
 	fmt.Fprintln(buf, `
 ifneq ($(my_check_failed),false)
-  $(error Soong variable check failed)
+  $(warning Soong variable check failed)
 endif
 my_check_failed :=
 
diff --git a/cc/config/arm_device.go b/cc/config/arm_device.go
index 1b60eae..d97e63e 100644
--- a/cc/config/arm_device.go
+++ b/cc/config/arm_device.go
@@ -51,10 +51,52 @@ var (
 		"-fgcse-after-reload",
 		"-frerun-cse-after-loop",
 		"-frename-registers",
+		"-DDISABLE_ASHMEM_TRACKING",
+		"-DUMOUNT_AND_FSCK_IS_UNSAFE",
+		"-DTARGET_LEGACY_CAMERA_HAL1",
+		"-DMETADATA_CAMERA_SOURCE",
+		"-DCAMCORDER_GRALLOC_SOURCE",
+		"-DHARDWARE_FIFO_SENSOR_SERVICE",
+		"-DCOMPAT_SENSORS_M",
+		"-DSTE_HARDWARE",
+		"-DSTE_SAMSUNG_HARDWARE",
+		"-DFORCE_SCREENSHOT_CPU_PATH",
+		"-DBOARD_CANT_REALLOCATE_OMX_BUFFERS",
+		"-DMR0_AUDIO_BLOB",
+		"-DMR1_AUDIO_BLOB",
+		"-DRECOVERY_CANT_USE_CONFIG_EXT4_FS_XATTR",
+		"-DNEEDS_VECTORIMPL_SYMBOLS",
+		"-DADD_LEGACY_ACQUIRE_BUFFER_SYMBOL",
+		"-DHAVE_PRE_KITKAT_AUDIO_POLICY_BLOB",
+		"-DHAVE_PRE_KITKAT_AUDIO_BLOB",
+		"-DCANNOT_SET_SURFACE_WITHOUT_A_FLUSH",
+		"-DREQUIRES_SYNCHRONOUS_SETSURFACE",
+		"-DEGL_NEEDS_HANDLE",
 	}
 
 	armCppflags = []string{
 		"-fvisibility-inlines-hidden",
+		"-DDISABLE_ASHMEM_TRACKING",
+		"-DUMOUNT_AND_FSCK_IS_UNSAFE",
+		"-DTARGET_LEGACY_CAMERA_HAL1",
+		"-DMETADATA_CAMERA_SOURCE",
+		"-DCAMCORDER_GRALLOC_SOURCE",
+		"-DHARDWARE_FIFO_SENSOR_SERVICE",
+		"-DCOMPAT_SENSORS_M",
+		"-DSTE_HARDWARE",
+		"-DSTE_SAMSUNG_HARDWARE",
+		"-DFORCE_SCREENSHOT_CPU_PATH",
+		"-DBOARD_CANT_REALLOCATE_OMX_BUFFERS",
+		"-DMR0_AUDIO_BLOB",
+		"-DMR1_AUDIO_BLOB",
+		"-DRECOVERY_CANT_USE_CONFIG_EXT4_FS_XATTR",
+		"-DNEEDS_VECTORIMPL_SYMBOLS",
+		"-DADD_LEGACY_ACQUIRE_BUFFER_SYMBOL",
+		"-DHAVE_PRE_KITKAT_AUDIO_POLICY_BLOB",
+		"-DHAVE_PRE_KITKAT_AUDIO_BLOB",
+		"-DCANNOT_SET_SURFACE_WITHOUT_A_FLUSH",
+		"-DREQUIRES_SYNCHRONOUS_SETSURFACE",
+		"-DEGL_NEEDS_HANDLE",
 	}
 
 	armLdflags = []string{
