Subject: [PATCH] AtomicFile: Rename the backup file to base file when backup exists

Restore the backup file as base file.

diff --git a/core/java/android/util/AtomicFile.java b/core/java/android/util/AtomicFile.java
index f573526d683..e4d2a4c1bbe 100644
--- a/core/java/android/util/AtomicFile.java
+++ b/core/java/android/util/AtomicFile.java
@@ -97,6 +97,7 @@ public FileOutputStream startWrite() throws IOException {
                 }
             } else {
                 mBaseName.delete();
+                mBackupName.renameTo(mBaseName);
             }
         }
         FileOutputStream str = null;
diff --git a/core/java/com/android/internal/os/AtomicFile.java b/core/java/com/android/internal/os/AtomicFile.java
index 5a83f33c875..3480c127f46 100644
--- a/core/java/com/android/internal/os/AtomicFile.java
+++ b/core/java/com/android/internal/os/AtomicFile.java
@@ -63,6 +63,7 @@ public FileOutputStream startWrite() throws IOException {
                 }
             } else {
                 mBaseName.delete();
+                mBackupName.renameTo(mBaseName);
             }
         }
         FileOutputStream str = null;
