Subject: [PATCH] fd_utils: don't whitelist /dev interfaces 
 add /dev/alarm to the whitelist.

diff --git a/core/jni/fd_utils.cpp b/core/jni/fd_utils.cpp
index e81a07e..7c20864 100644
--- a/core/jni/fd_utils.cpp
+++ b/core/jni/fd_utils.cpp
@@ -130,20 +130,12 @@ FileDescriptorInfo* FileDescriptorInfo::CreateFromFd(int fd) {
     return NULL;
   }
 
-  const FileDescriptorWhitelist* whitelist = FileDescriptorWhitelist::Get();
-
   if (S_ISSOCK(f_stat.st_mode)) {
     std::string socket_name;
     if (!GetSocketName(fd, &socket_name)) {
       return NULL;
     }
 
-    if (!whitelist->IsAllowed(socket_name)) {
-      LOG(ERROR) << "Socket name not whitelisted : " << socket_name
-                 << " (fd=" << fd << ")";
-      return NULL;
-    }
-
     return new FileDescriptorInfo(fd);
   }
 
@@ -168,11 +160,6 @@ FileDescriptorInfo* FileDescriptorInfo::CreateFromFd(int fd) {
     return NULL;
   }
 
-  if (!whitelist->IsAllowed(file_path)) {
-    LOG(ERROR) << "Not whitelisted : " << file_path;
-    return NULL;
-  }
-
   // File descriptor flags : currently on FD_CLOEXEC. We can set these
   // using F_SETFD - we're single threaded at this point of execution so
   // there won't be any races.
diff --git a/core/jni/fd_utils.cpp b/core/jni/fd_utils.cpp
index e81a07e..7020864 100644
--- a/core/jni/fd_utils.cpp
+++ b/core/jni/fd_utils.cpp
@@ -33,6 +33,7 @@
 
 // Static whitelist of open paths that the zygote is allowed to keep open.
 static const char* kPathWhitelist[] = {
+  "/dev/alarm",
   "/dev/null",
   "/dev/socket/zygote",
   "/dev/socket/zygote_secondary",
