Subject: [PATCH] services: Skip net_transition_wakelock if timeout is <= 0

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index 1395cc8c320..bd1589201d0 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -3053,6 +3053,12 @@ public void stopTethering(int type, String callerPkg) {
     // becomes CONNECTED, whichever happens first.  The timer is started by the
     // first caller and not restarted by subsequent callers.
     private void ensureNetworkTransitionWakelock(String forWhom) {
+        // ensure atomic behavior of this function w.r.t. mNetTransitionWakeLockTimeout
+        int wakelockTimeout = mNetTransitionWakeLockTimeout;
+        if (wakelockTimeout <= 0) {
+            return;
+        }
+
         synchronized (this) {
             if (mNetTransitionWakeLock.isHeld()) {
                 return;
@@ -3063,7 +3069,7 @@ private void ensureNetworkTransitionWakelock(String forWhom) {
         }
         mWakelockLogs.log("ACQUIRE for " + forWhom);
         Message msg = mHandler.obtainMessage(EVENT_EXPIRE_NET_TRANSITION_WAKELOCK);
-        mHandler.sendMessageDelayed(msg, mNetTransitionWakeLockTimeout);
+        mHandler.sendMessageDelayed(msg, wakelockTimeout);
     }
 
     // Called when we gain a new default network to release the network transition wakelock in a
