Subject: [PATCH] InputWrapper: Opt out early if session == null

diff --git a/core/java/android/inputmethodservice/IInputMethodWrapper.java b/core/java/android/inputmethodservice/IInputMethodWrapper.java
index 765aff9..43581d8 100644
--- a/core/java/android/inputmethodservice/IInputMethodWrapper.java
+++ b/core/java/android/inputmethodservice/IInputMethodWrapper.java
@@ -268,6 +268,10 @@
 
     @Override
     public void setSessionEnabled(IInputMethodSession session, boolean enabled) {
+        if (session == null) {
+            Log.w(TAG, "Incoming session is null");
+            return;
+        }
         try {
             InputMethodSession ls = ((IInputMethodSessionWrapper)
                     session).getInternalInputMethodSession();
