Subject: [PATCH] base: SettingsLib: stop ClassCastException

diff --git a/packages/SettingsLib/src/com/android/settingslib/suggestions/SuggestionParser.java b/packages/SettingsLib/src/com/android/settingslib/suggestions/SuggestionParser.java
index 56b84415e907..1a64c7d41099 100644
--- a/packages/SettingsLib/src/com/android/settingslib/suggestions/SuggestionParser.java
+++ b/packages/SettingsLib/src/com/android/settingslib/suggestions/SuggestionParser.java
@@ -392,7 +392,11 @@ private String getDismissControl(Tile suggestion, boolean isSmartSuggestionEnabl
         if (isSmartSuggestionEnabled) {
             return mDefaultDismissControl;
         } else {
-            return suggestion.metaData.getString(META_DATA_DISMISS_CONTROL);
+            Object o = suggestion.metaData.get(META_DATA_DISMISS_CONTROL);
+            if (o instanceof String) {
+                return (String) o;
+            }
+            return null;
         }
     }
 
