Add debug ACodec to allow using Treble 

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 830de19..01eea89 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -576,7 +576,7 @@ ACodec::ACodec()
 
     changeState(mUninitializedState);
 
-    mTrebleFlag = false;
+    mTrebleFlag = bool(property_get_bool("debug.treble_omx", false));
 }
 
 ACodec::~ACodec() {
