Subject: [PATCH] Fix graphicBuffer

diff --git a/libs/gui/BufferQueueProducer.cpp b/libs/gui/BufferQueueProducer.cpp
index e60a39a..320b27f 100644
--- a/libs/gui/BufferQueueProducer.cpp
+++ b/libs/gui/BufferQueueProducer.cpp
@@ -515,15 +515,15 @@ status_t BufferQueueProducer::dequeueBuffer(int *outSlot,
         { // Autolock scope
             Mutex::Autolock lock(mCore->mMutex);
 
-            if (error == NO_ERROR && !mCore->mIsAbandoned) {
+            if (graphicBuffer != NULL && !mCore->mIsAbandoned) {
                 graphicBuffer->setGenerationNumber(mCore->mGenerationNumber);
                 mSlots[*outSlot].mGraphicBuffer = graphicBuffer;
             }
 
             mCore->mIsAllocating = false;
             mCore->mIsAllocatingCondition.broadcast();
 
-            if (error != NO_ERROR) {
+            if (graphicBuffer == NULL) {
                 mCore->mFreeSlots.insert(*outSlot);
                 mCore->clearBufferSlotLocked(*outSlot);
                 BQ_LOGE("dequeueBuffer: createGraphicBuffer failed");
