 Remove -ffast-math from libhwui makefile

bug:25417885

diff --git a/libs/hwui/Android.common.mk b/libs/hwui/Android.common.mk
index 00f7bb2..08fc477 100644
--- a/libs/hwui/Android.common.mk
+++ b/libs/hwui/Android.common.mk
@@ -119,7 +119,7 @@ endif
 # Defaults for ATRACE_TAG and LOG_TAG for libhwui
 LOCAL_CFLAGS += -DATRACE_TAG=ATRACE_TAG_VIEW -DLOG_TAG=\"OpenGLRenderer\"
 LOCAL_CFLAGS += -Wall -Wno-unused-parameter -Wunreachable-code
-LOCAL_CFLAGS += -ffast-math -O3
+LOCAL_CFLAGS += -O3
 
 # b/21698669
 ifneq ($(USE_CLANG_PLATFORM_BUILD),true)
