diff --git a/service/java/com/android/server/wifi/WifiScanningServiceImpl.java b/service/java/com/android/server/wifi/WifiScanningServiceImpl.java
index 08f67d4..7f25772 100755
--- a/service/java/com/android/server/wifi/WifiScanningServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiScanningServiceImpl.java
@@ -1013,6 +1013,8 @@ private int getBestBucket(ScanSettings settings) {
                     break;
                 }
             }
+            if (bestBucketIndex > mSettings.buckets.length)
+                bestBucketIndex = -1;
 
             if (mostFrequentBucketIndex < bestBucketIndex) {
                 for (ChannelSpec desiredChannelSpec : channels) {
