SHack: fix wifi autoconnection [2/3]

diff --git a/core/java/android/os/SystemClock.java b/core/java/android/os/SystemClock.java
index 00a5803..3b82593 100644
--- a/core/java/android/os/SystemClock.java
+++ b/core/java/android/os/SystemClock.java
@@ -175,6 +175,13 @@ public static boolean setCurrentTimeMillis(long millis) {
     @CriticalNative
     native public static long elapsedRealtime();
 
+    /**
+     * Returns milliseconds since boot, including time spent in sleep.
+     *
+     * @return elapsed milliseconds since boot.
+     */
+    native public static long elapsedRealtime1();
+
     /**
      * Returns nanoseconds since boot, including time spent in sleep.
      *
@@ -183,6 +190,13 @@ public static boolean setCurrentTimeMillis(long millis) {
     @CriticalNative
     public static native long elapsedRealtimeNanos();
 
+    /**
+     * Returns nanoseconds since boot, including time spent in sleep.
+     *
+     * @return elapsed nanoseconds since boot.
+     */
+    public static native long elapsedRealtimeNanos1();
+
     /**
      * Returns milliseconds running in the current thread.
      *
diff --git a/core/jni/android_os_SystemClock.cpp b/core/jni/android_os_SystemClock.cpp
index 00d6dc7..f56b619 100644
--- a/core/jni/android_os_SystemClock.cpp
+++ b/core/jni/android_os_SystemClock.cpp
@@ -88,6 +88,8 @@ static const JNINativeMethod gMethods[] = {
     { "uptimeMillis", "()J", (void*) uptimeMillis },
     { "elapsedRealtime", "()J", (void*) elapsedRealtime },
     { "elapsedRealtimeNanos", "()J", (void*) elapsedRealtimeNano },
+    { "elapsedRealtime1", "()J", (void*) elapsedRealtime1 },
+    { "elapsedRealtimeNanos1", "()J", (void*) elapsedRealtimeNano1 },
 
     // SystemClock doesn't have an implementation for these that we can directly call
     { "currentThreadTimeMillis", "()J",
