Subject: [PATCH] Add NewSettings proto

diff --git a/proto/src/metrics_constants.proto b/proto/src/metrics_constants.proto
index 81a23f0afa3..6de16e95a93 100644
--- a/proto/src/metrics_constants.proto
+++ b/proto/src/metrics_constants.proto
@@ -4604,5 +4604,8 @@ message MetricsEvent {
 
     // Add new aosp constants above this line.
     // END OF AOSP CONSTANTS
+
+    // NewSettings constant
+    NEW_SETTINGS = 1500;
   }
 }
