Subject: [PATCH] fwb: Keyguard: Hide check button when using PIN quick unlock

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java b/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
index 0097a29653d..7c531286efc 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
@@ -161,6 +161,8 @@ public void onValidateQuickUnlock(String password) {
         } else {
             mPasswordEntry.setQuickUnlockListener(null);
         }
+
+        setButtonVisibility(getOkButton(), !quickUnlock);
     }
 
     @Override
diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardPinBasedInputView.java b/packages/SystemUI/src/com/android/keyguard/KeyguardPinBasedInputView.java
index c04ae68dab3..6dca81714aa 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardPinBasedInputView.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardPinBasedInputView.java
@@ -51,6 +51,17 @@ public KeyguardPinBasedInputView(Context context, AttributeSet attrs) {
         super(context, attrs);
     }
 
+    public void setButtonVisibility(View button, boolean visible) {
+        if (button instanceof View && button != null) {
+            button.setVisibility(
+                    visible ? View.VISIBLE : View.INVISIBLE);
+        }
+    }
+
+    public View getOkButton() {
+        return mOkButton;
+    }
+
     @Override
     public void reset() {
         mPasswordEntry.requestFocus();
diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardSimPinView.java b/packages/SystemUI/src/com/android/keyguard/KeyguardSimPinView.java
index 2e36130853c..2a253d4746c 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardSimPinView.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardSimPinView.java
@@ -167,7 +167,9 @@ protected void onFinishInflate() {
             ((EmergencyCarrierArea) mEcaView).setCarrierTextVisible(true);
         }
         mSimImageView = findViewById(R.id.keyguard_sim);
+
         mPasswordEntry.setQuickUnlockListener(null);
+        setButtonVisibility(getOkButton(), true /* visible */);
     }
 
     @Override
diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardSimPukView.java b/packages/SystemUI/src/com/android/keyguard/KeyguardSimPukView.java
index 52dc2200818..eaedc46bc96 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardSimPukView.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardSimPukView.java
@@ -216,7 +216,9 @@ protected void onFinishInflate() {
             ((EmergencyCarrierArea) mEcaView).setCarrierTextVisible(true);
         }
         mSimImageView = findViewById(R.id.keyguard_sim);
+
         mPasswordEntry.setQuickUnlockListener(null);
+        setButtonVisibility(getOkButton(), true);
     }
 
     @Override
