Subject: [PATCH] fwb: Add ability to permanently hide apps from recents [1/2]

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 7281854ee7a..954a3734eb8 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -4024,6 +4024,12 @@ public boolean validate(String value) {
         public static final String TOAST_ANIMATION = "toast_animation";
 
         /**
+         * List of Apps hidden from recents
+         * @hide
+         */
+        public static final String HIDE_FROM_RECENTS_LIST = "hide_from_recents_list";
+
+        /**
          * Settings to backup. This is here so that it's in the same place as the settings
          * keys and easy to update.
          *
diff --git a/services/core/java/com/android/server/am/ActivityRecord.java b/services/core/java/com/android/server/am/ActivityRecord.java
index 55c4698528a..bcde50c90e9 100755
--- a/services/core/java/com/android/server/am/ActivityRecord.java
+++ b/services/core/java/com/android/server/am/ActivityRecord.java
@@ -126,6 +126,8 @@
 import android.app.PictureInPictureParams;
 import android.app.ResultInfo;
 import android.content.ComponentName;
+import android.content.ContentResolver;
+import android.content.ContextWrapper;
 import android.content.Intent;
 import android.content.pm.ActivityInfo;
 import android.content.pm.ApplicationInfo;
@@ -147,6 +149,7 @@
 import android.os.UserHandle;
 import android.os.storage.StorageManager;
 import android.service.voice.IVoiceInteractionSession;
+import android.provider.Settings;
 import android.util.EventLog;
 import android.util.Log;
 import android.util.MergedConfiguration;
@@ -882,6 +885,29 @@ boolean isResolverActivity() {
             realTheme = aInfo.applicationInfo.targetSdkVersion < HONEYCOMB
                     ? android.R.style.Theme : android.R.style.Theme_Holo;
         }
+
+        String pkgName = aInfo.packageName;
+        String hideFromRecentsString = Settings.System.getStringForUser(service.mContext.getContentResolver(),
+                Settings.System.HIDE_FROM_RECENTS_LIST, UserHandle.USER_CURRENT);
+        ArrayList<String> excludeFromRecentsList = new ArrayList();
+
+        // this converts the String we get from Settings to an actual ArrayList
+        if (hideFromRecentsString!=null && hideFromRecentsString.length()!=0){
+            String[] parts = hideFromRecentsString.split("\\|");
+            for(int i = 0; i < parts.length; i++){
+                excludeFromRecentsList.add(parts[i]);
+            }
+        }
+
+        if (!excludeFromRecentsList.isEmpty()){
+            if (excludeFromRecentsList.contains(pkgName)) {
+              // If our app is inside the ArrayList, hide it from the Recents.
+              // For the case where that flag already was added by some other instance,
+              // it most likely has a good reason to be, so do not force remove it
+              intent.addFlags(Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
+            }
+        }
+
         if ((aInfo.flags & ActivityInfo.FLAG_HARDWARE_ACCELERATED) != 0) {
             windowFlags |= LayoutParams.FLAG_HARDWARE_ACCELERATED;
         }
