diff --git a/core/java/android/app/ApplicationPackageManager.java b/core/java/android/app/ApplicationPackageManager.java
index db4e123..e0b5320 100644
--- a/core/java/android/app/ApplicationPackageManager.java
+++ b/core/java/android/app/ApplicationPackageManager.java
@@ -1643,19 +1643,28 @@
 
     @Override
     public @NonNull List<VolumeInfo> getPackageCandidateVolumes(ApplicationInfo app) {
+        return getPackageCandidateVolumesInternal(app, false);
+    }
+
+    @Override
+    public @NonNull List<VolumeInfo> getPackageCandidateVolumesForceable(ApplicationInfo app) {
+        return getPackageCandidateVolumesInternal(app, true);
+    }
+
+    private @NonNull List<VolumeInfo> getPackageCandidateVolumesInternal(ApplicationInfo app, boolean forceable) {
         final StorageManager storage = mContext.getSystemService(StorageManager.class);
         final VolumeInfo currentVol = getPackageCurrentVolume(app);
         final List<VolumeInfo> vols = storage.getVolumes();
         final List<VolumeInfo> candidates = new ArrayList<>();
         for (VolumeInfo vol : vols) {
-            if (Objects.equals(vol, currentVol) || isPackageCandidateVolume(app, vol)) {
+            if (Objects.equals(vol, currentVol) || isPackageCandidateVolume(app, vol, forceable)) {
                 candidates.add(vol);
             }
         }
         return candidates;
     }
 
-    private static boolean isPackageCandidateVolume(ApplicationInfo app, VolumeInfo vol) {
+    private static boolean isPackageCandidateVolume(ApplicationInfo app, VolumeInfo vol, boolean forceable) {
         // Private internal is always an option
         if (VolumeInfo.ID_PRIVATE_INTERNAL.equals(vol.getId())) {
             return true;
@@ -1665,7 +1674,7 @@
         // anywhere else
         if (app.isSystemApp()
                 || app.installLocation == PackageInfo.INSTALL_LOCATION_INTERNAL_ONLY
-                || app.installLocation == PackageInfo.INSTALL_LOCATION_UNSPECIFIED) {
+                || (!forceable && app.installLocation == PackageInfo.INSTALL_LOCATION_UNSPECIFIED)) {
             return false;
         }
 
diff --git a/core/java/android/content/pm/PackageManager.java b/core/java/android/content/pm/PackageManager.java
index 529d641..64416f1 100644
--- a/core/java/android/content/pm/PackageManager.java
+++ b/core/java/android/content/pm/PackageManager.java
@@ -4524,6 +4524,8 @@
     public abstract @Nullable VolumeInfo getPackageCurrentVolume(ApplicationInfo app);
     /** {@hide} */
     public abstract @NonNull List<VolumeInfo> getPackageCandidateVolumes(ApplicationInfo app);
+    /** {@hide} */
+    public abstract @NonNull List<VolumeInfo> getPackageCandidateVolumesForceable(ApplicationInfo app);
 
     /** {@hide} */
     public abstract int movePrimaryStorage(VolumeInfo vol);
diff --git a/test-runner/src/android/test/mock/MockPackageManager.java b/test-runner/src/android/test/mock/MockPackageManager.java
index d3e2bfd..6742cfa 100644
--- a/test-runner/src/android/test/mock/MockPackageManager.java
+++ b/test-runner/src/android/test/mock/MockPackageManager.java
@@ -594,6 +594,12 @@
 
     /** {@hide} */
     @Override
+    public List<VolumeInfo> getPackageCandidateVolumesForceable(ApplicationInfo app) {
+        throw new UnsupportedOperationException();
+    }
+
+    /** {@hide} */
+    @Override
     public int movePrimaryStorage(VolumeInfo vol) {
         throw new UnsupportedOperationException();
     }
