Bring back UMS support

diff --git a/packages/SettingsLib/res/values-ru/arrays.xml b/packages/SettingsLib/res/values-ru/arrays.xml
index 1110081..2889013 100644
--- a/packages/SettingsLib/res/values-ru/arrays.xml
+++ b/packages/SettingsLib/res/values-ru/arrays.xml
@@ -253,5 +253,6 @@
     <item msgid="7398830860950841822">"RNDIS (USB Ethernet)"</item>
     <item msgid="1718924214939774352">"Источник аудио"</item>
     <item msgid="8126315616613006284">"MIDI"</item>
+    <item>UMS (USB Mass Storage)</item>
   </string-array>
 </resources>
diff --git a/packages/SettingsLib/res/values/arrays.xml b/packages/SettingsLib/res/values/arrays.xml
index 1110071..2889013 100644
--- a/packages/SettingsLib/res/values/arrays.xml
+++ b/packages/SettingsLib/res/values/arrays.xml
@@ -509,6 +509,7 @@
         <item>RNDIS (USB Ethernet)</item>
         <item>Audio Source</item>
         <item>MIDI</item>
+        <item>UMS (USB Mass Storage)</item>
     </string-array>
 
     <!-- USB configuration values for Developer Settings.
@@ -528,6 +529,8 @@
         <item>audio_source</item>
         <!-- Do not translate. -->
         <item>midi</item>
+        <!-- Do not translate. -->
+        <item>mass_storage</item>
     </string-array>
 
     <!-- Display color space adjustment modes for developers -->
diff --git a/core/java/android/hardware/usb/UsbManager.java b/core/java/android/hardware/usb/UsbManager.java
index 059a7a9..7e3439c 100644
--- a/core/java/android/hardware/usb/UsbManager.java
+++ b/core/java/android/hardware/usb/UsbManager.java
@@ -226,6 +226,14 @@
     public static final String USB_FUNCTION_MTP = "mtp";
 
     /**
+     * Name of the UMS USB function.
+     * Used in extras for the {@link #ACTION_USB_STATE} broadcast
+     *
+     * {@hide}
+     */
+    public static final String USB_FUNCTION_UMS = "mass_storage";
+
+    /**
      * Name of the PTP USB function.
      * Used in extras for the {@link #ACTION_USB_STATE} broadcast
      *
diff --git a/services/usb/java/com/android/server/usb/UsbService.java b/services/usb/java/com/android/server/usb/UsbService.java
index 059a7a9..7e34390 100644
--- a/services/usb/java/com/android/server/usb/UsbService.java
+++ b/services/usb/java/com/android/server/usb/UsbService.java
@@ -399,6 +399,7 @@
             case UsbManager.USB_FUNCTION_MTP:
             case UsbManager.USB_FUNCTION_PTP:
             case UsbManager.USB_FUNCTION_RNDIS:
+            case UsbManager.USB_FUNCTION_UMS:
                 return true;
         }
 
diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 059a7a9..7e34391 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -1055,7 +1055,8 @@
 
         private boolean isUsbDataTransferActive() {
             return UsbManager.containsFunction(mCurrentFunctions, UsbManager.USB_FUNCTION_MTP)
-                    || UsbManager.containsFunction(mCurrentFunctions, UsbManager.USB_FUNCTION_PTP);
+                    || UsbManager.containsFunction(mCurrentFunctions, UsbManager.USB_FUNCTION_PTP)
+                    || UsbManager.containsFunction(mCurrentFunctions, UsbManager.USB_FUNCTION_UMS);
         }
 
         public UsbAccessory getCurrentAccessory() {
@@ -1115,6 +1116,10 @@
                         UsbManager.USB_FUNCTION_MIDI)) {
                     titleRes = com.android.internal.R.string.usb_midi_notification_title;
                     id = SystemMessage.NOTE_USB_MIDI;
+                } else if (UsbManager.containsFunction(mCurrentFunctions,
+                        UsbManager.USB_FUNCTION_UMS)) {
+                    titleRes = com.android.internal.R.string.usb_mtp_notification_title;
+                    id = SystemMessage.NOTE_USB_UMS;
                 } else if (UsbManager.containsFunction(mCurrentFunctions,
                         UsbManager.USB_FUNCTION_ACCESSORY)) {
                     titleRes = com.android.internal.R.string.usb_accessory_notification_title;
diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 059a709..7e34391 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -701,7 +701,8 @@
 
                 if (mBootCompleted
                         && (UsbManager.containsFunction(functions, UsbManager.USB_FUNCTION_MTP)
-                        || UsbManager.containsFunction(functions, UsbManager.USB_FUNCTION_PTP))) {
+                        || UsbManager.containsFunction(functions, UsbManager.USB_FUNCTION_PTP)
+                        || UsbManager.containsFunction(functions, UsbManager.USB_FUNCTION_UMS))) {
                     // Start up dependent services.
                     updateUsbStateBroadcastIfNeeded(true);
                 }
diff --git a/proto/src/system_messages.proto b/proto/src/system_messages.proto
index 81a23f0afa3..60e16e95a93 100644
--- a/proto/src/system_messages.proto
+++ b/proto/src/system_messages.proto
@@ -191,6 +191,10 @@ message MetricsEvent {
     // Legacy IDs with arbitrary values appear below
     // Legacy IDs existed as stable non-conflicting constants prior to the O release
 
+    // Inform that USB is configured for USB Mass Storage
+    // Package: android
+    NOTE_USB_UMS = 50;
+
     // Network status notes, previously decleared in metrics_constants with these values
     // Package: android
     //
