Subject: [PATCH] wifi: set user root

diff --git a/wifi/1.1/default/android.hardware.wifi@1.0-service.rc b/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
index 770c830..3edbda6 100644
--- a/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
+++ b/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
@@ -1,4 +1,4 @@
 service wifi_hal_legacy /vendor/bin/hw/android.hardware.wifi@1.0-service
     class hal
-    user wifi
+    user root
     group wifi gps
