diff --git a/Android.mk b/Android.mk
index 770d2b6..00fc5fa 100644
--- a/Android.mk
+++ b/Android.mk
@@ -14,6 +14,8 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE_TAGS := optional
 
+LOCAL_CFLAGS += -Os
+
 LOCAL_SRC_FILES := \
         $(call all-java-files-under, src)
 
diff --git a/jni/Android.mk b/jni/Android.mk
index 780d2b6..00fc5fa 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -32,7 +32,7 @@ LOCAL_SHARED_LIBRARIES := \
 
 LOCAL_MULTILIB := 32
 
-#LOCAL_CFLAGS += -O0 -g
+LOCAL_CFLAGS += -Os
 
 LOCAL_MODULE := libbluetooth_jni
 LOCAL_MODULE_TAGS := optional
