Subject: [PATCH] Setting:BugFix for OOM caused by looper leak in settings

diff --git a/src/com/android/settings/dashboard/DashboardFragment.java b/src/com/android/settings/dashboard/DashboardFragment.java
index 8c3ac0d026..5bd6776461 100644
--- a/src/com/android/settings/dashboard/DashboardFragment.java
+++ b/src/com/android/settings/dashboard/DashboardFragment.java
@@ -369,4 +369,11 @@ void refreshDashboardTiles(final String TAG) {
         }
         mSummaryLoader.setListening(true);
     }
+
+    public void onDestroy(){
+        if(mSummaryLoader != null){
+           mSummaryLoader.release();
+        }
+        super.onDestroy();
+    }
 }
