Bring back UMS support [1/3]

diff --git a/src/com/android/settings/deviceinfo/UsbBackend.java b/src/com/android/settings/deviceinfo/UsbBackend.java
index 1110072..2889014 100644
--- a/src/com/android/settings/deviceinfo/UsbBackend.java
+++ b/src/com/android/settings/deviceinfo/UsbBackend.java
@@ -37,6 +37,7 @@
     public static final int MODE_DATA_MTP    = 0x01 << 1;
     public static final int MODE_DATA_PTP    = 0x02 << 1;
     public static final int MODE_DATA_MIDI   = 0x03 << 1;
+    public static final int MODE_DATA_UMS    = 0x04 << 1;
 
     private final boolean mRestricted;
     private final boolean mRestrictedBySystem;
@@ -96,6 +97,8 @@ public int getUsbDataMode() {
             return MODE_DATA_PTP;
         } else if (mUsbManager.isFunctionEnabled(UsbManager.USB_FUNCTION_MIDI)) {
             return MODE_DATA_MIDI;
+        } else if (mUsbManager.isFunctionEnabled(UsbManager.USB_FUNCTION_UMS)) {
+            return MODE_DATA_UMS;
         }
         return MODE_DATA_NONE; // ...
     }
@@ -118,6 +121,9 @@ private void setUsbFunction(int mode) {
             case MODE_DATA_MIDI:
                 mUsbManager.setCurrentFunction(UsbManager.USB_FUNCTION_MIDI, true);
                 break;
+            case MODE_DATA_UMS:
+                mUsbManager.setCurrentFunction(UsbManager.USB_FUNCTION_UMS, true);
+                break;
             default:
                 mUsbManager.setCurrentFunction(null, false);
                 break;
diff --git a/src/com/android/settings/deviceinfo/UsbModeChooserActivity.java b/src/com/android/settings/deviceinfo/UsbModeChooserActivity.java
index 1110073..2889015 100644
--- a/src/com/android/settings/deviceinfo/UsbModeChooserActivity.java
+++ b/src/com/android/settings/deviceinfo/UsbModeChooserActivity.java
@@ -55,7 +55,8 @@
         UsbBackend.MODE_POWER_SOURCE | UsbBackend.MODE_DATA_NONE,
         UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_MTP,
         UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_PTP,
-        UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_MIDI
+        UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_MIDI,
+        UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_UMS
     };
 
     private UsbBackend mBackend;
@@ -198,6 +199,8 @@ private static int getTitle(int mode) {
                 return R.string.usb_use_photo_transfers;
             case UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_MIDI:
                 return R.string.usb_use_MIDI;
+            case UsbBackend.MODE_POWER_SINK | UsbBackend.MODE_DATA_UMS:
+                return R.string.usb_use_UMS;
         }
         return 0;
     }
diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
index 022f220..7289093 100644
--- a/res/values/cr_strings.xml
+++ b/res/values/cr_strings.xml
@@ -24,4 +24,7 @@
     <string name="allow_signature_fake_summary">Allow apps to bypass security systems by pretending to be a different app</string>
     <string name="allow_signature_fake_warning">Allowing apps to bypass security systems can lead to serious security and privacy problems! Check that only benign apps use the corresponding permission when this is active.</string>
 
+    <!-- transferring files via UMS -->
+    <string name="usb_use_UMS">File transfers via UMS</string>
+
 </resources>
