Subject: [PATCH] init: fix reboot to recovery / bootloader

diff --git a/init/reboot.cpp b/init/reboot.cpp
index aac1b5c17..78140e752 100644
--- a/init/reboot.cpp
+++ b/init/reboot.cpp
@@ -75,6 +75,10 @@ enum UmountStat {
     UMOUNT_STAT_NOT_AVAILABLE = 4,
 };
 
+std::string param_mnt_dir("/mnt/.lfs");
+std::string ramdisk_mnt_dir("/ramdisk");
+std::string system_mnt_dir("/system");
+
 // Utility for struct mntent
 class MountEntry {
   public:
@@ -85,7 +89,17 @@ class MountEntry {
           mnt_opts_(entry.mnt_opts) {}
 
     bool Umount(bool force) {
-        int r = umount2(mnt_dir_.c_str(), force ? MNT_FORCE : 0);
+        int r;
+
+        if (mnt_dir_.compare(param_mnt_dir) == 0 ||
+            mnt_dir_.compare(ramdisk_mnt_dir) == 0 ||
+            mnt_dir_.compare(system_mnt_dir) == 0)
+        {
+            LOG(INFO) << "skipping umount of " << mnt_fsname_ << ":" << mnt_dir_ << " opts " << mnt_opts_;
+            return true;
+        }
+
+        r = umount2(mnt_dir_.c_str(), force ? MNT_FORCE : 0);
         if (r == 0) {
             LOG(INFO) << "umounted " << mnt_fsname_ << ":" << mnt_dir_ << " opts " << mnt_opts_;
             return true;
