diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index 00550c3..6a687e1 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -207,27 +207,15 @@ status_t PublicVolume::doMount() {
     dev_t before = GetDevice(mFuseWrite);
 
     if (!(mFusePid = fork())) {
-        if (getMountFlags() & MountFlags::kPrimary) {
-            if (execl(kFusePath, kFusePath,
-                    "-u", "1023", // AID_MEDIA_RW
-                    "-g", "1023", // AID_MEDIA_RW
-                    "-U", std::to_string(getMountUserId()).c_str(),
-                    "-w",
-                    mRawPath.c_str(),
-                    stableName.c_str(),
-                    NULL)) {
-                PLOG(ERROR) << "Failed to exec";
-            }
-        } else {
-            if (execl(kFusePath, kFusePath,
-                    "-u", "1023", // AID_MEDIA_RW
-                    "-g", "1023", // AID_MEDIA_RW
-                    "-U", std::to_string(getMountUserId()).c_str(),
-                    mRawPath.c_str(),
-                    stableName.c_str(),
-                    NULL)) {
-                PLOG(ERROR) << "Failed to exec";
-            }
+        if (execl(kFusePath, kFusePath,
+                  "-u", "1023", // AID_MEDIA_RW
+                  "-g", "1023", // AID_MEDIA_RW
+                  "-U", std::to_string(getMountUserId()).c_str(),
+                  "-w",
+                  mRawPath.c_str(),
+                  stableName.c_str(),
+                  NULL)) {
+            PLOG(ERROR) << "Failed to exec";
         }
 
         LOG(ERROR) << "FUSE exiting";
diff --git a/VolumeManager.cpp b/VolumeManager.cpp
index 01550c3..6a687e1 100644
--- a/VolumeManager.cpp
+++ b/VolumeManager.cpp
@@ -283,8 +283,8 @@ void VolumeManager::handleBlockEvent(NetlinkEvent *evt) {
         evt->dump();
     }
 
-    std::string eventPath(evt->findParam("DEVPATH"));
-    std::string devType(evt->findParam("DEVTYPE"));
+    std::string eventPath(evt->findParam("DEVPATH")?evt->findParam("DEVPATH"):"");
+    std::string devType(evt->findParam("DEVTYPE")?evt->findParam("DEVTYPE"):"");
 
     if (devType != "disk") return;
 
diff --git a/fs/Vfat.cpp b/fs/Vfat.cpp
index 02550c3..6a687e1 100644
--- a/fs/Vfat.cpp
+++ b/fs/Vfat.cpp
@@ -108,6 +108,11 @@ status_t Check(const std::string& source) {
             errno = EIO;
             return -1;
 
+        case 8:
+            SLOGE("Filesystem check failed (no filesystem)");
+            errno = ENODATA;
+            return -1;
+
         default:
             SLOGE("Filesystem check failed (unknown exit code %d)", rc);
             errno = EIO;
