diff --git a/init/init.cpp b/init/init.cpp
index 320d2ba..a3fc5fa 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -936,14 +936,6 @@ static bool selinux_is_disabled(void)
     return false;
 }
 
-static bool selinux_is_enforcing(void)
-{
-    if (ALLOW_DISABLE_SELINUX) {
-        return selinux_status_from_cmdline() == SELINUX_ENFORCING;
-    }
-    return true;
-}
-
 int selinux_reload_policy(void)
 {
     if (selinux_is_disabled()) {
@@ -997,7 +989,7 @@ static void selinux_initialize(bool in_kernel_domain) {
             security_failure();
         }
 
-        bool is_enforcing = selinux_is_enforcing();
+        bool is_enforcing = false;
         security_setenforce(is_enforcing);
 
         if (write_file("/sys/fs/selinux/checkreqprot", "0") == -1) {
diff --git a/init/Android.mk b/init/Android.mk
index 320d2ba..13fc5fa 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -4,12 +4,7 @@ LOCAL_PATH:= $(call my-dir)
 
 # --
 
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
 init_options += -DALLOW_LOCAL_PROP_OVERRIDE=1 -DALLOW_DISABLE_SELINUX=1
-else
-init_options += -DALLOW_LOCAL_PROP_OVERRIDE=0 -DALLOW_DISABLE_SELINUX=0
-endif
-
 init_options += -DLOG_UEVENTS=0
 
 init_cflags += \
