Don't start tethering if IPv6 RPF is not supported. 

diff --git a/server/NatController.cpp b/server/NatController.cpp
index cbc3611..4129c0f 100644
--- a/server/NatController.cpp
+++ b/server/NatController.cpp
@@ -205,7 +205,7 @@
         "*raw\n"
         "%s %s -i %s -m rpfilter --invert ! -s fe80::/64 -j DROP\n"
         "COMMIT\n", op, LOCAL_RAW_PREROUTING, intIface);
-    if (iptablesRestoreFunction(V6, rpfilterCmd) == -1 && add) {
+    if (iptablesRestoreFunction(V6, rpfilterCmd) == -1 && add && false /* STOPSHIP */) {
         return -1;
     }
 
