Subject: [PATCH] Bootloop fixes

diff --git a/init/Android.mk b/init/Android.mk
index 6e79dbd..a5b3eee 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -8,7 +8,7 @@
 init_options += \
     -DALLOW_LOCAL_PROP_OVERRIDE=1 \
     -DALLOW_PERMISSIVE_SELINUX=1 \
-    -DREBOOT_BOOTLOADER_ON_PANIC=1 \
+    -DREBOOT_BOOTLOADER_ON_PANIC=0 \
     -DDUMP_ON_UMOUNT_FAILURE=1
 else
 init_options += \
diff --git a/init/init.cpp b/init/init.cpp
index db47920..972a544 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -430,8 +430,8 @@ static int set_mmap_rnd_bits_action(const std::vector<std::string>& args)
 #endif
 
     if (ret == -1) {
-        LOG(ERROR) << "Unable to set adequate mmap entropy value!";
-        security_failure();
+        LOG(INFO) << "Unable to set adequate mmap entropy value!";
+        // security_failure();
     }
     return ret;
 }
@@ -977,6 +977,8 @@ static void install_reboot_signal_handlers() {
 
         // panic() reboots to bootloader
         panic();
+        // panic1() reboots to recovery
+        panic1();
     };
     action.sa_flags = SA_RESTART;
     sigaction(SIGABRT, &action, nullptr);
diff --git a/init/util.cpp b/init/util.cpp
index dfa6c62..a939948 100644
--- a/init/util.cpp
+++ b/init/util.cpp
@@ -376,6 +376,12 @@ void panic() {
     DoReboot(ANDROID_RB_RESTART2, "reboot", "bootloader", false);
 }
 
+void panic1() {
+    LOG(ERROR) << "panic: rebooting to recovery";
+    // Do not queue "shutdown" trigger since we want to shutdown immediately
+    DoReboot(ANDROID_RB_RESTART2, "reboot", "recovery", false);
+}
+
 static std::string init_android_dt_dir() {
     // Use the standard procfs-based path by default
     std::string android_dt_dir = kDefaultAndroidDtDir;
diff --git a/init/util.h b/init/util.h
index dfa6c82..a939948 100644
--- a/init/util.h
+++ b/init/util.h
@@ -54,6 +54,7 @@
 bool expand_props(const std::string& src, std::string* dst);
 
 void panic() __attribute__((__noreturn__));
+void panic1() __attribute__((__noreturn__));
 
 // Returns the platform's Android DT directory as specified in the kernel cmdline.
 // If the platform does not configure a custom DT path, returns the standard one (based in procfs).
