diff --git a/config/common.mk b/config/common.mk
index 2fed3dd..448cec7 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -152,9 +152,8 @@
     CMFileManager \
     Eleven \
     LockClock \
-    CMUpdater \
     CyanogenSetupWizard \
     CMSettingsProvider \
     ExactCalculator
 
 # CM Platform Library
diff --git a/config/common_full.mk b/config/common_full.mk
index 2fed3d0..448cec7 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -15,8 +15,7 @@ PRODUCT_PACKAGES += \
     PhaseBeam \
     PhotoTable \
     SoundRecorder \
-    PhotoPhase \
-    Screencast
+    PhotoPhase
 
 # Extra tools in CM
 PRODUCT_PACKAGES += \
