From 66a98283afeb1efe23cbeb0455925404075bc8f6 Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Tue, 4 Sep 2018 13:50:56 +0200
Subject: [PATCH 01/11] VoldNativeService: Allow underscore for ID

Change-Id: I1775c3b6f5875a355a31653932d3d826f03cdd1f
---
 VoldNativeService.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/VoldNativeService.cpp b/VoldNativeService.cpp
index 81523c6..4d7efa3 100644
--- a/VoldNativeService.cpp
+++ b/VoldNativeService.cpp
@@ -106,7 +106,7 @@ binder::Status checkArgumentId(const std::string& id) {
         return exception(binder::Status::EX_ILLEGAL_ARGUMENT, "Missing ID");
     }
     for (const char& c : id) {
-        if (!std::isalnum(c) && c != ':' && c != ',') {
+        if (!std::isalnum(c) && c != ':' && c != ',' && c != '_') {
             return exception(binder::Status::EX_ILLEGAL_ARGUMENT,
                     StringPrintf("ID %s is malformed", id.c_str()));
         }
-- 
2.11.0

