From 08909780fade66ac35536c73f00f4e8ff4560bbc Mon Sep 17 00:00:00 2001
From: morckx <morckx@gmail.com>
Date: Tue, 10 Nov 2015 09:12:42 +0100
Subject: [PATCH 01/10] Use volume label for mount path, if available.

---
 PublicVolume.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index 1a0a538..b191aea 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -116,9 +116,11 @@ status_t PublicVolume::doMount() {
         return -EIO;
     }
 
-    // Use UUID as stable name, if available
+    // Use volume label and otherwise UUID as stable name, if available
     std::string stableName = getId();
-    if (!mFsUuid.empty()) {
+    if (!mFsLabel.empty()) {
+        stableName = mFsLabel;
+    } else if (!mFsUuid.empty()) {
         stableName = mFsUuid;
     }
 
-- 
2.11.0

