From 943122873a335f49f8c2da93cf5ff121c990122b Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 04/10] Make public volumes publicly writable based on
 WRITE_EXTERNAL_STORAGE

---
 PublicVolume.cpp | 31 ++++++++++---------------------
 1 file changed, 10 insertions(+), 21 deletions(-)

diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index 7d29eaa..c8070dd 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -220,27 +220,16 @@ status_t PublicVolume::doMount() {
     dev_t before = GetDevice(mFuseWrite);
 
     if (!(mFusePid = fork())) {
-        if (getMountFlags() & MountFlags::kPrimary) {
-            if (execl(kFusePath, kFusePath,
-                    "-u", "1023", // AID_MEDIA_RW
-                    "-g", "1023", // AID_MEDIA_RW
-                    "-U", std::to_string(getMountUserId()).c_str(),
-                    "-w",
-                    mRawPath.c_str(),
-                    stableName.c_str(),
-                    NULL)) {
-                PLOG(ERROR) << "Failed to exec";
-            }
-        } else {
-            if (execl(kFusePath, kFusePath,
-                    "-u", "1023", // AID_MEDIA_RW
-                    "-g", "1023", // AID_MEDIA_RW
-                    "-U", std::to_string(getMountUserId()).c_str(),
-                    mRawPath.c_str(),
-                    stableName.c_str(),
-                    NULL)) {
-                PLOG(ERROR) << "Failed to exec";
-            }
+        if (execl(kFusePath, kFusePath,
+                  "-u", "1023", // AID_MEDIA_RW
+                  "-g", "1023", // AID_MEDIA_RW
+                  "-U", std::to_string(getMountUserId()).c_str(),
+                  "-m",
+                  "-w",
+                  mRawPath.c_str(),
+                  stableName.c_str(),
+                  NULL)) {
+            PLOG(ERROR) << "Failed to exec";
         }
 
         LOG(ERROR) << "FUSE exiting";
-- 
2.11.0

