From 7d0e359260ab42484279b00888db671ba6e2d023 Mon Sep 17 00:00:00 2001
From: Shilin Victor <radicaldreamer00001@gmail.com>
Date: Fri, 29 Mar 2019 22:55:05 +0300
Subject: [PATCH 7/9] system-vold_W06-dont-override-mFsLabel.patch

Change-Id: I5b4f0cbe70f2f6422e902b20a9815485618791c1
---
 PublicVolume.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index c30f8b5..ad8902b 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -62,10 +62,10 @@ PublicVolume::~PublicVolume() {
 }
 
 status_t PublicVolume::readMetadata() {
-    status_t res = ReadMetadataUntrusted(mDevPath, mFsType, mFsUuid, mFsLabel);
+    std::string unused;
+    status_t res = ReadMetadataUntrusted(mDevPath, mFsType, mFsUuid, unused);
     notifyEvent(ResponseCode::VolumeFsTypeChanged, mFsType);
     notifyEvent(ResponseCode::VolumeFsUuidChanged, mFsUuid);
-    notifyEvent(ResponseCode::VolumeFsLabelChanged, mFsLabel);
     return res;
 }
 
-- 
2.11.0

