From 17699ccd1cae9f1dce412da8db409e897c2e86ce Mon Sep 17 00:00:00 2001
From: Shilin Victor <radicaldreamer00001@gmail.com>
Date: Fri, 29 Mar 2019 22:55:06 +0300
Subject: [PATCH 9/9] system-vold_W08-Add-creating-crypt-block-device.patch

Change-Id: I081a3bd27d24d4b499791cface9f00855328d491
---
 cryptfs.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/cryptfs.c b/cryptfs.c
index 7a5c80e..696624b 100644
--- a/cryptfs.c
+++ b/cryptfs.c
@@ -1358,6 +1358,8 @@ static int create_crypto_blk_dev(struct crypt_mnt_ftr *crypt_ftr,
   int version[3];
   char *extra_params;
   int load_count;
+  struct stat statbuf;
+  int retry = 25;
 #ifdef CONFIG_HW_DISK_ENCRYPTION
   char encrypted_state[PROPERTY_VALUE_MAX] = {0};
   char progress[PROPERTY_VALUE_MAX] = {0};
@@ -1437,6 +1439,16 @@ static int create_crypto_blk_dev(struct crypt_mnt_ftr *crypt_ftr,
     goto errout;
   }
 
+  /* wait for creating dm device */
+  while (stat(crypto_blk_name, &statbuf) != 0) {
+      retry--;
+      if (retry < 0) {
+          SLOGE("Timed out to getting dm blkdev status.\n");
+          goto errout;
+      }
+      usleep(40 * 1000);
+  }
+
   /* We made it here with no errors.  Woot! */
   retval = 0;
 
-- 
2.11.0

