From c67346dbd1fb921ed8af8eaa3cfbd3ee0531ed0e Mon Sep 17 00:00:00 2001
From: morckx <morckx@gmail.com>
Date: Tue, 10 Nov 2015 09:12:42 +0100
Subject: [PATCH 21/27] Use volume label for mount path, if available.

---
 model/PublicVolume.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/model/PublicVolume.cpp b/model/PublicVolume.cpp
index b19ab8a..f842697 100644
--- a/model/PublicVolume.cpp
+++ b/model/PublicVolume.cpp
@@ -106,9 +106,11 @@ status_t PublicVolume::doMount() {
         return -EIO;
     }
 
-    // Use UUID as stable name, if available
+    // Use volume label and otherwise UUID as stable name, if available
     std::string stableName = getId();
-    if (!mFsUuid.empty()) {
+    if (!mFsLabel.empty()) {
+        stableName = mFsLabel;
+    } else if (!mFsUuid.empty()) {
         stableName = mFsUuid;
     }
 
-- 
2.11.0

