From d1401026faf6473a365f388af044a929ba0dd2de Mon Sep 17 00:00:00 2001
From: morckx <morckx@gmail.com>
Date: Tue, 10 Nov 2015 09:12:42 +0100
Subject: [PATCH] Use volume label for mount path, if available.

diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index eb550c3..6a687e1 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -116,9 +116,11 @@
         return -EIO;
     }
 
-    // Use UUID as stable name, if available
+    // Use volume label and otherwise UUID as stable name, if available
     std::string stableName = getId();
-    if (!mFsUuid.empty()) {
+    if (!mFsLabel.empty()) {
+        stableName = mFsLabel;
+    } else if (!mFsUuid.empty()) {
         stableName = mFsUuid;
     }
 
