From 432d8dcae8526bb914f1592ac23bede0d62efe05 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@gmail.com>
Date: Sun, 8 Nov 2015 01:55:19 +0200
Subject: [PATCH] busybox: fix building android-8.0

---
 Android.mk | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index d74efad8e..1ef6fbead 100644
--- a/Android.mk
+++ b/Android.mk
@@ -7,8 +7,8 @@ BB_PATH := $(LOCAL_PATH)
 BIONIC_ICS := false
 BIONIC_L := true
 
-BUSYBOX_WARNING_HIDE := -Wno-error=implicit-function-declaration -Wno-implicit-function-declaration -Wno-implicit-fallthrough \
-			-Wno-sign-compare -Wno-format-overflow -Wno-shift-negative-value -Wno-logical-not-parentheses -Wno-return-type
+BUSYBOX_WARNING_HIDE := -Wno-error=implicit-function-declaration -Wno-implicit-function-declaration \
+			-Wno-sign-compare -Wno-return-type
 
 # Make a static library for regex.
 include $(CLEAR_VARS)
@@ -24,7 +24,6 @@ LOCAL_SRC_FILES := $(shell cat $(BB_PATH)/android/librpc.sources)
 LOCAL_C_INCLUDES := $(BB_PATH)/android/librpc
 LOCAL_MODULE := libuclibcrpc
 LOCAL_CFLAGS += -fno-strict-aliasing $(BUSYBOX_WARNING_HIDE)
-LOCAL_CLANG := false
 ifeq ($(BIONIC_L),true)
 LOCAL_CFLAGS += -DBIONIC_ICS -DBIONIC_L
 endif
-- 
2.11.0

