diff --git a/Android.mk b/Android.mk
index 27b67c8..fd27971 100644
--- a/Android.mk
+++ b/Android.mk
@@ -86,9 +86,9 @@ BUSYBOX_C_INCLUDES = \
 	bionic/libm/include \
 	bionic/libc \
 	bionic/libm \
+        external/libselinux/include \
+        external/selinux/libsepol/include \
 	libc/kernel/common \
-	external/libselinux/include \
-	external/selinux/libsepol/include \
 	$(BB_PATH)/android/regex \
 	$(BB_PATH)/android/librpc
 
@@ -132,7 +132,7 @@ LOCAL_CFLAGS += \
 LOCAL_ASFLAGS := $(BUSYBOX_AFLAGS)
 LOCAL_MODULE := libbusybox
 LOCAL_MODULE_TAGS := eng debug
-LOCAL_STATIC_LIBRARIES := libcutils libc libm libselinux
+LOCAL_STATIC_LIBRARIES := libcutils libc libm
 LOCAL_ADDITIONAL_DEPENDENCIES := $(busybox_prepare_minimal)
 include $(BUILD_STATIC_LIBRARY)
 
@@ -152,7 +152,7 @@ LOCAL_MODULE := busybox
 LOCAL_MODULE_TAGS := eng debug
 LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
 LOCAL_SHARED_LIBRARIES := libc libcutils libm
-LOCAL_STATIC_LIBRARIES := libclearsilverregex libuclibcrpc libselinux
+LOCAL_STATIC_LIBRARIES := libclearsilverregex libuclibcrpc
 LOCAL_ADDITIONAL_DEPENDENCIES := $(busybox_prepare_full)
 LOCAL_CLANG := false
 include $(BUILD_EXECUTABLE)
@@ -201,7 +201,7 @@ LOCAL_FORCE_STATIC_EXECUTABLE := true
 LOCAL_MODULE := static_busybox
 LOCAL_MODULE_STEM := busybox
 LOCAL_MODULE_TAGS := optional
-LOCAL_STATIC_LIBRARIES := libclearsilverregex libc libcutils libm libuclibcrpc libselinux
+LOCAL_STATIC_LIBRARIES := libclearsilverregex libc libcutils libm libuclibcrpc
 LOCAL_MODULE_CLASS := UTILITY_EXECUTABLES
 LOCAL_MODULE_PATH := $(PRODUCT_OUT)/utilities
 LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/utilities
@@ -229,7 +229,7 @@ LOCAL_FORCE_STATIC_EXECUTABLE := true
 LOCAL_MODULE := static_min_busybox
 LOCAL_MODULE_STEM := busyboxmin
 LOCAL_MODULE_TAGS := optional
-LOCAL_STATIC_LIBRARIES := libclearsilverregex libc libcutils libm libuclibcrpc libselinux
+LOCAL_STATIC_LIBRARIES := libclearsilverregex libc libcutils libm libuclibcrpc
 LOCAL_MODULE_CLASS := UTILITY_EXECUTABLES
 LOCAL_MODULE_PATH := $(PRODUCT_OUT)/utilities
 LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/utilities
diff --git a/busybox-full.config b/busybox-full.config
index b11099b..46ea06b 100644
--- a/busybox-full.config
+++ b/busybox-full.config
@@ -46,7 +46,7 @@ CONFIG_PID_FILE_PATH=""
 CONFIG_FEATURE_SUID=y
 # CONFIG_FEATURE_SUID_CONFIG is not set
 # CONFIG_FEATURE_SUID_CONFIG_QUIET is not set
-CONFIG_SELINUX=y
+CONFIG_SELINUX=n
 # CONFIG_FEATURE_PREFER_APPLETS is not set
 CONFIG_BUSYBOX_EXEC_PATH="/proc/self/exe"
 CONFIG_FEATURE_SYSLOG=y
@@ -167,7 +167,7 @@ CONFIG_FEATURE_TAR_LONG_OPTIONS=y
 CONFIG_FEATURE_TAR_TO_COMMAND=y
 CONFIG_FEATURE_TAR_UNAME_GNAME=y
 CONFIG_FEATURE_TAR_NOPRESERVE_TIME=y
-CONFIG_FEATURE_TAR_SELINUX=y
+CONFIG_FEATURE_TAR_SELINUX=n
 CONFIG_UNZIP=y
 
 #
@@ -949,25 +949,6 @@ CONFIG_SV_DEFAULT_SERVICE_DIR=""
 # CONFIG_SOFTLIMIT is not set
 
 #
-# SELinux Utilities
-#
-CONFIG_CHCON=y
-CONFIG_FEATURE_CHCON_LONG_OPTIONS=y
-CONFIG_GETENFORCE=y
-CONFIG_GETSEBOOL=y
-# CONFIG_LOAD_POLICY is not set
-CONFIG_MATCHPATHCON=y
-CONFIG_RESTORECON=y
-CONFIG_RUNCON=y
-CONFIG_FEATURE_RUNCON_LONG_OPTIONS=y
-CONFIG_SELINUXENABLED=y
-CONFIG_SETENFORCE=y
-CONFIG_SETFILES=y
-# CONFIG_FEATURE_SETFILES_CHECK_OPTION is not set
-CONFIG_SETSEBOOL=y
-CONFIG_SESTATUS=y
-
-#
 # Shells
 #
 CONFIG_ASH=y
diff --git a/busybox-full.sources b/busybox-full.sources
index bd5b59c..4a0ddf1 100644
--- a/busybox-full.sources
+++ b/busybox-full.sources
@@ -69,11 +69,6 @@ procps/lsof.c procps/nmeter.c procps/pstree.c procps/pwdx.c procps/smemcap.c
 
 shell/ash.c shell/ash_ptr_hack.c shell/math.c shell/random.c shell/shell_common.c
 
-libbb/selinux_common.c android/selinux/matchpathcon.c android/selinux/stubs.c
-selinux/chcon.c selinux/selinuxenabled.c
-selinux/getenforce.c selinux/sestatus.c selinux/setsebool.c
-selinux/getsebool.c selinux/runcon.c selinux/setenforce.c selinux/setfiles.c selinux/matchpathcon.c
-
 util-linux/blkid.c util-linux/blockdev.c util-linux/dmesg.c util-linux/fdisk.c util-linux/findfs.c util-linux/flock.c
 util-linux/freeramdisk.c util-linux/fstrim.c util-linux/getopt.c
 util-linux/hexdump.c util-linux/hwclock.c util-linux/losetup.c util-linux/lspci.c util-linux/lsusb.c
diff --git a/busybox-minimal.config b/busybox-minimal.config
index eeed3bb..243c6b2 100644
--- a/busybox-minimal.config
+++ b/busybox-minimal.config
@@ -46,7 +46,7 @@ CONFIG_PID_FILE_PATH=""
 CONFIG_FEATURE_SUID=y
 # CONFIG_FEATURE_SUID_CONFIG is not set
 # CONFIG_FEATURE_SUID_CONFIG_QUIET is not set
-CONFIG_SELINUX=y
+CONFIG_SELINUX=n
 # CONFIG_FEATURE_PREFER_APPLETS is not set
 CONFIG_BUSYBOX_EXEC_PATH="/proc/self/exe"
 # CONFIG_FEATURE_SYSLOG is not set
@@ -167,7 +167,7 @@ CONFIG_FEATURE_TAR_LONG_OPTIONS=y
 # CONFIG_FEATURE_TAR_TO_COMMAND is not set
 # CONFIG_FEATURE_TAR_UNAME_GNAME is not set
 CONFIG_FEATURE_TAR_NOPRESERVE_TIME=y
-CONFIG_FEATURE_TAR_SELINUX=y
+CONFIG_FEATURE_TAR_SELINUX=n
 CONFIG_UNZIP=y
 
 #
@@ -944,25 +944,6 @@ CONFIG_SV_DEFAULT_SERVICE_DIR=""
 # CONFIG_SOFTLIMIT is not set
 
 #
-# SELinux Utilities
-#
-CONFIG_CHCON=y
-# CONFIG_FEATURE_CHCON_LONG_OPTIONS is not set
-CONFIG_GETENFORCE=y
-CONFIG_GETSEBOOL=y
-# CONFIG_LOAD_POLICY is not set
-CONFIG_MATCHPATHCON=y
-CONFIG_RESTORECON=y
-# CONFIG_RUNCON is not set
-# CONFIG_FEATURE_RUNCON_LONG_OPTIONS is not set
-CONFIG_SELINUXENABLED=y
-CONFIG_SETENFORCE=y
-CONFIG_SETFILES=y
-# CONFIG_FEATURE_SETFILES_CHECK_OPTION is not set
-CONFIG_SETSEBOOL=y
-CONFIG_SESTATUS=y
-
-#
 # Shells
 #
 CONFIG_ASH=y
diff --git a/busybox-minimal.sources b/busybox-minimal.sources
index 84950cf..b5abaf2 100644
--- a/busybox-minimal.sources
+++ b/busybox-minimal.sources
@@ -44,11 +44,6 @@ procps/renice.c procps/sysctl.c procps/top.c procps/uptime.c procps/watch.c
 
 shell/ash.c shell/ash_ptr_hack.c shell/math.c shell/random.c shell/shell_common.c
 
-android/selinux/matchpathcon.c android/selinux/stubs.c
-libbb/selinux_common.c selinux/chcon.c selinux/selinuxenabled.c
-selinux/getenforce.c selinux/sestatus.c selinux/setsebool.c
-selinux/getsebool.c selinux/setenforce.c selinux/setfiles.c selinux/matchpathcon.c
-
 util-linux/blkid.c util-linux/blockdev.c util-linux/dmesg.c util-linux/fdisk.c util-linux/freeramdisk.c util-linux/fstrim.c util-linux/getopt.c
 util-linux/hexdump.c util-linux/losetup.c util-linux/lspci.c util-linux/lsusb.c util-linux/mkfs_ext2.c util-linux/mkfs_vfat.c util-linux/mkswap.c
 util-linux/more.c util-linux/mount.c util-linux/rdev.c util-linux/rev.c util-linux/swaponoff.c util-linux/switch_root.c util-linux/umount.c
