From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
diff --git a/charger/Android.mk b/charger/Android.mk
index 9fed3dd..4480ec7 100644
--- a/charger/Android.mk
+++ b/charger/Android.mk
@@ -1,4 +1,5 @@
-LOCAL_PATH := $(call my-dir)
+ifeq ($(WITH_LINEAGE_CHARGER),true)
+LOCAL_PATH := $(call my-dir)
 
 # Set healthd_density to the density bucket of the device.
 healthd_density := unknown
@@ -68,4 +69,5 @@
 include $(BUILD_PHONY_PACKAGE)
 
 _add-charger-image :=
-_img_modules :=
+_img_modules :=
+endif
