From 98cc7c21ac28e0f6f135354bba7faad45c45e5f3 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 26 Dec 2016 18:48:09 +0700
Subject: [PATCH 1/4] vendor-cm_device-scheduler.patch

---
 prebuilt/common/etc/init.local.rc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/prebuilt/common/etc/init.local.rc b/prebuilt/common/etc/init.local.rc
index 101280e..cc4ad03 100644
--- a/prebuilt/common/etc/init.local.rc
+++ b/prebuilt/common/etc/init.local.rc
@@ -99,8 +99,11 @@ on boot
     chmod 0664 /sys/devices/system/cpu/cpufreq/ondemand/up_threshold_any_cpu_load
 
     chown system system /sys/block/mmcblk0/queue/scheduler
+    chown system system /sys/block/mmcblk1/queue/scheduler
     chmod 0664 /sys/block/mmcblk0/queue/scheduler
+    chmod 0664 /sys/block/mmcblk1/queue/scheduler
     restorecon /sys/block/mmcblk0/queue/scheduler
+    restorecon /sys/block/mmcblk1/queue/scheduler
 
     chown system system /sys/block/sda/queue/scheduler
     chmod 0664 /sys/block/sda/queue/scheduler
-- 
2.9.3

