From 50d87de678067a1089ef9dbef4d099ac7186e212 Mon Sep 17 00:00:00 2001
From: Ningyuan Wang <nywang@google.com>
Date: Mon, 3 Apr 2017 10:26:27 -0700
Subject: [PATCH] wificond: set user root

diff --git a/wificond.rc b/wificond.rc
index c7e1b43..236f700 100644
--- a/wificond.rc
+++ b/wificond.rc
@@ -1,4 +1,4 @@ service wificond /system/bin/wificond
 service wificond /system/bin/wificond
     class main
-    user wifi
+    user root
     group wifi net_raw net_admin
