From 3800210deb59ea2f548d452870bd869aa96cb107 Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Sun, 6 Nov 2016 01:49:51 -0800
Subject: [PATCH] telephony: Fix NPE when switching data SIM

diff --git a/src/java/com/android/internal/telephony/Phone.java b/src/java/com/android/internal/telephony/Phone.java
index 0341d3a..e6168cd 100644
--- a/src/java/com/android/internal/telephony/Phone.java
+++ b/src/java/com/android/internal/telephony/Phone.java
@@ -3477,7 +3477,9 @@ public void registerForDataEnabledChanged(Handler h, int what, Object obj) {
     }
 
     public void unregisterForDataEnabledChanged(Handler h) {
-        mDcTracker.unregisterForDataEnabledChanged(h);
+        if (mDcTracker != null) {
+            mDcTracker.unregisterForDataEnabledChanged(h);
+        }
     }
 
     public IccSmsInterfaceManager getIccSmsInterfaceManager(){
