From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Fix wifi off
diff --git a/service/java/com/android/server/wifi/WifiStateMachine.java b/service/java/com/android/server/wifi/WifiStateMachine.java
index 08f67d4..7f25772 100755
--- a/service/java/com/android/server/wifi/WifiStateMachine.java
+++ b/service/java/com/android/server/wifi/WifiStateMachine.java
@@ -3914,7 +3914,10 @@ public class WifiStateMachine extends StateMachine {
                     maybeRegisterPhoneListener();
                     break;
                 case CMD_SCREEN_STATE_CHANGED:
-                    handleScreenStateChanged(message.arg1 != 0);
+                    WifiManager manager = (WifiManager)mContext.getSystemService(Context.WIFI_SERVICE);
+                    if(manager.isWifiEnabled()) {
+                        handleScreenStateChanged(message.arg1 != 0);
+                    }
                     break;
                     /* Discard */
                 case CMD_START_SCAN:
