 Revert "egl: Remove old tuna BGRA vs RGBA workaround." 

diff --git a/opengl/libs/Android.bp b/opengl/libs/Android.bp
index d823938..992c1fe 100644
--- a/opengl/libs/Android.bp
+++ b/opengl/libs/Android.bp
@@ -122,6 +122,13 @@ cc_library_static {
 
 cc_library_shared {
     name: "libEGL",
+    product_variables: {
+        lineage: {
+            needs_egl_khr_workaround: {
+                cflags: ["-DWORKAROUND_BUG_10194508=1"],
+            },
+        },
+    },
     defaults: ["egl_libs_defaults"],
     srcs: [
         "EGL/egl_tls.cpp",
diff --git a/opengl/libs/EGL/eglApi.cpp b/opengl/libs/EGL/eglApi.cpp
index d823938..992c1fe 100644
--- a/opengl/libs/EGL/eglApi.cpp
+++ b/opengl/libs/EGL/eglApi.cpp
@@ -617,6 +617,14 @@ void getNativePixelFormat(EGLDisplay dpy, egl_connection_t* cnx, EGLConfig confi
     EGLint componentType = EGL_COLOR_COMPONENT_TYPE_FIXED_EXT;
     cnx->egl.eglGetConfigAttrib(dpy, config, EGL_COLOR_COMPONENT_TYPE_EXT, &componentType);
 
+#if WORKAROUND_BUG_10194508
+    if (!cnx->egl.eglGetConfigAttrib(dpy, config, EGL_NATIVE_VISUAL_ID,
+            &format)) {
+        ALOGE("eglGetConfigAttrib(EGL_NATIVE_VISUAL_ID) failed: %#x",
+                eglGetError());
+        format = 0;
+    }
+#else
     EGLint a = 0;
     EGLint r, g, b;
     r = g = b = 0;
@@ -664,6 +672,7 @@ void getNativePixelFormat(EGLDisplay dpy, egl_connection_t* cnx, EGLConfig confi
             format = HAL_PIXEL_FORMAT_RGBA_FP16;
         }
     }
+#endif
 }
 
 EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
