diff --git a/libs/gui/Android.bp b/libs/gui/Android.bp
index 9770da2..033e764 100644
--- a/libs/gui/Android.bp
+++ b/libs/gui/Android.bp
@@ -27,7 +27,6 @@
     clang: true,
     cppflags: [
         "-Weverything",
-        "-Werror",
 
         // The static constructors and destructors in this library have not been noted to
         // introduce significant overheads
diff --git a/services/surfaceflinger/Android.mk b/services/surfaceflinger/Android.mk
index 9770da2..033e764 100644
--- a/services/surfaceflinger/Android.mk
+++ b/services/surfaceflinger/Android.mk
@@ -105,7 +105,7 @@
     libhidltransport \
     libhwbinder
 
-LOCAL_CFLAGS += -Wall -Werror -Wunused -Wunreachable-code -std=c++1z
+LOCAL_CFLAGS += -Wall -Wunused -Wunreachable-code -std=c++1z
 
 include $(BUILD_SHARED_LIBRARY)
 
@@ -154,7 +154,7 @@
 LOCAL_32_BIT_ONLY := true
 endif
 
-LOCAL_CFLAGS += -Wall -Werror -Wunused -Wunreachable-code
+LOCAL_CFLAGS += -Wall -Wunused -Wunreachable-code
 
 include $(BUILD_EXECUTABLE)
 
@@ -177,7 +177,7 @@
 
 LOCAL_MODULE := libsurfaceflinger_ddmconnection
 
-LOCAL_CFLAGS += -Wall -Werror -Wunused -Wunreachable-code
+LOCAL_CFLAGS += -Wall -Wunused -Wunreachable-code
 
 include $(BUILD_SHARED_LIBRARY)
 endif # libnativehelper
diff --git a/libs/ui/Android.bp b/libs/ui/Android.bp
index 9770da2..033e764 100644
--- a/libs/ui/Android.bp
+++ b/libs/ui/Android.bp
@@ -22,7 +22,6 @@
     clang: true,
     cppflags: [
         "-Weverything",
-        "-Werror",
 
         // The static constructors and destructors in this library have not been noted to
         // introduce significant overheads
diff --git a/libs/sensor/Android.bp b/libs/sensor/Android.bp
index 87e5b45..96341b0 100644
--- a/libs/sensor/Android.bp
+++ b/libs/sensor/Android.bp
@@ -18,7 +18,6 @@
     clang: true,
     cppflags: [
         "-Weverything",
-        "-Werror",
 
         // The static constructors and destructors in this library have not been noted to
         // introduce significant overheads
diff --git a/libs/binder/Android.bp b/libs/binder/Android.bp
index 9770da2..033e769 100644
--- a/libs/binder/Android.bp
+++ b/libs/binder/Android.bp
@@ -79,7 +79,6 @@
     cflags: [
         "-Wall",
         "-Wextra",
-        "-Werror",
     ],
     product_variables: {
         binder32bit: {
