diff --git a/src/Android.mk b/src/Android.mk
index 4f3f665..14990b6 100644
--- a/src/Android.mk
+++ b/src/Android.mk
@@ -14,6 +14,7 @@ LOCAL_CFLAGS += -fvisibility=hidden -UNDEBUG
 LOCAL_CFLAGS += -Wall -Werror
 
 LOCAL_MODULE := libOpenSLESUT
+LOCAL_CLANG := true
 
 include $(BUILD_STATIC_LIBRARY)
 
@@ -33,7 +34,7 @@ LOCAL_SRC_FILES:=                     \
         assert.c \
         MPH_to.c \
         handlers.c
-
+LOCAL_CLANG := true
 LOCAL_MODULE:= libopensles_helper
 
 include $(BUILD_STATIC_LIBRARY)
@@ -199,6 +200,7 @@ LOCAL_SHARED_LIBRARIES += \
         libstagefright_http_support
 endif
 
+LOCAL_CLANG := true
 LOCAL_MODULE := libwilhelm
 LOCAL_MODULE_TAGS := optional
 
@@ -216,6 +218,7 @@ LOCAL_C_INCLUDES:=                                                  \
         frameworks/av/media/libstagefright/include                \
         frameworks/native/include/media/openmax
 LOCAL_MODULE := libOpenSLES
+LOCAL_CLANG := true
 LOCAL_MODULE_TAGS := optional
 LOCAL_CFLAGS += -x c++ -std=gnu++11 -DLI_API= -fvisibility=hidden -UNDEBUG \
                 -DSL_API='__attribute__((visibility("default")))'
@@ -231,6 +234,7 @@ LOCAL_C_INCLUDES:=                                                  \
         frameworks/av/media/libstagefright/include                \
         frameworks/native/include/media/openmax
 LOCAL_MODULE := libOpenMAXAL
+LOCAL_CLANG := true
 LOCAL_MODULE_TAGS := optional
 LOCAL_CFLAGS += -x c++ -std=gnu++11 -DLI_API= -fvisibility=hidden -UNDEBUG \
                 -DXA_API='__attribute__((visibility("default")))'
diff --git a/src/Android.mk b/src/Android.mk
index 4f3f665..14990b6 100644
--- a/src/Android.mk
+++ b/src/Android.mk
@@ -14,6 +14,7 @@ LOCAL_CFLAGS += -fvisibility=hidden -UNDEBUG
 LOCAL_CFLAGS += -Wall -Werror
 
 LOCAL_MODULE := libOpenSLESUT
+LOCAL_CLANG := true
 
 include $(BUILD_STATIC_LIBRARY)
 
@@ -33,7 +34,7 @@ LOCAL_SRC_FILES:=                     \
         assert.c \
         MPH_to.c \
         handlers.c
-
+LOCAL_CLANG := true
 LOCAL_MODULE:= libopensles_helper
 
 include $(BUILD_STATIC_LIBRARY)
@@ -199,6 +200,7 @@ LOCAL_SHARED_LIBRARIES += \
         libstagefright_http_support
 endif
 
+LOCAL_CLANG := true
 LOCAL_MODULE := libwilhelm
 LOCAL_MODULE_TAGS := optional
 
@@ -216,6 +218,7 @@ LOCAL_C_INCLUDES:=                                                  \
         frameworks/av/media/libstagefright/include                \
         frameworks/native/include/media/openmax
 LOCAL_MODULE := libOpenSLES
+LOCAL_CLANG := true
 LOCAL_MODULE_TAGS := optional
 LOCAL_CFLAGS += -x c++ -std=gnu++11 -DLI_API= -fvisibility=hidden -UNDEBUG \
                 -DSL_API='__attribute__((visibility("default")))'
@@ -231,6 +234,7 @@ LOCAL_C_INCLUDES:=                                                  \
         frameworks/av/media/libstagefright/include                \
         frameworks/native/include/media/openmax
 LOCAL_MODULE := libOpenMAXAL
+LOCAL_CLANG := true
 LOCAL_MODULE_TAGS := optional
 LOCAL_CFLAGS += -x c++ -std=gnu++11 -DLI_API= -fvisibility=hidden -UNDEBUG \
                 -DXA_API='__attribute__((visibility("default")))'
