stagefright: render video buffers 

diff --git a/cmds/stagefright/Android.mk b/cmds/stagefright/Android.mk
index 64c7edc..7008078 100644
--- a/cmds/stagefright/Android.mk
+++ b/cmds/stagefright/Android.mk
@@ -9,7 +9,7 @@ LOCAL_SRC_FILES:=       \
 
 LOCAL_SHARED_LIBRARIES := \
 	libstagefright libmedia libutils libbinder libstagefright_foundation \
-        libjpeg libgui libcutils liblog
+        libjpeg libgui libcutils liblog libui
 
 LOCAL_C_INCLUDES:= \
 	frameworks/av/media/libstagefright \
diff --git a/cmds/stagefright/stagefright.cpp b/cmds/stagefright/stagefright.cpp
index 64c7edc..8008078 100644
--- a/cmds/stagefright/stagefright.cpp
+++ b/cmds/stagefright/stagefright.cpp
@@ -342,6 +342,12 @@ static void playSource(OMXClient *client, sp<MediaSource> &source) {
                     printf(".");
                     fflush(stdout);
                 }
+
+                // render buffers from OMXCodec
+                if (buffer->graphicBuffer() != NULL && gSurface != NULL) {
+                    gSurface->queueBuffer(gSurface.get(), buffer->graphicBuffer()->getNativeBuffer(), -1);
+                    buffer->meta_data()->setInt32(kKeyRendered, 1);
+                }
             }
 
             sumDecodeUs += delayDecodeUs;
