Revert "optimize wallpaper load,avoid show black wallpaper." 

diff --git a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
index e9ace29..0e51958 100644
--- a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
+++ b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
@@ -236,9 +236,7 @@ public void onVisibilityChanged(boolean visible) {
                     Log.d(TAG, "Visibility changed to visible=" + visible);
                 }
                 mVisible = visible;
-                if (visible) {
-                    drawFrame();
-                }
+                drawFrame();
             }
         }
 
