Replace AppOpsManager locking with std::lock_guard 

diff --git a/libs/binder/AppOpsManager.cpp b/libs/binder/AppOpsManager.cpp
index 90e5b4d..9634db7 100644
--- a/libs/binder/AppOpsManager.cpp
+++ b/libs/binder/AppOpsManager.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <mutex>
 #include <binder/AppOpsManager.h>
 #include <binder/Binder.h>
 #include <binder/IServiceManager.h>
@@ -41,8 +42,8 @@ AppOpsManager::AppOpsManager()
 
 sp<IAppOpsService> AppOpsManager::getService()
 {
+    std::lock_guard<Mutex> scoped_lock(mLock);
     int64_t startTime = 0;
-    mLock.lock();
     sp<IAppOpsService> service = mService;
     while (service == NULL || !IInterface::asBinder(service)->isBinderAlive()) {
         sp<IBinder> binder = defaultServiceManager()->checkService(_appops);
@@ -62,7 +63,6 @@ sp<IAppOpsService> AppOpsManager::getService()
             mService = service;
         }
     }
-    mLock.unlock();
     return service;
 }
 
