Init and Make USB Configuration option persistent 

diff --git a/src/com/android/settings/DevelopmentSettings.java b/src/com/android/settings/DevelopmentSettings.java
index 0005524..00ec124 100644
--- a/src/com/android/settings/DevelopmentSettings.java
+++ b/src/com/android/settings/DevelopmentSettings.java
@@ -1612,8 +1612,10 @@ private void updateUsbConfigurationValues() {
             String[] titles = getResources().getStringArray(R.array.usb_configuration_titles);
             int index = 0;
             // Assume if !isUnlocked -> charging, which should be at index 0
-            for (int i = 0; i < titles.length && isUnlocked; i++) {
-                if (manager.isFunctionEnabled(values[i])) {
+            String currentSetValue = SystemProperties.get("persist.usbconfiguration", "mtp");
+
+            for (int i = 0; i < values.length && isUnlocked; i++) {
+                if (currentSetValue.equals(values[i])) {
                     index = i;
                     break;
                 }
@@ -1627,6 +1629,8 @@ private void updateUsbConfigurationValues() {
     private void writeUsbConfigurationOption(Object newValue) {
         UsbManager manager = (UsbManager)getActivity().getSystemService(Context.USB_SERVICE);
         String function = newValue.toString();
+        SystemProperties.set("persist.usbconfiguration", function);
+
         if (function.equals("none")) {
             manager.setCurrentFunction(null);
             manager.setUsbDataUnlocked(false);
