diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
new file mode 100644
index 01034ba..6f30d6d
--- /dev/null
+++ b/res/values/cr_strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2012-2015 SlimRoms Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+
+    <!-- Title for the option managing whether or not to use intrusive incall UI. -->
+    <string name="use_intrusive_call_title">Intrusive incoming calls</string>
+    <string name="use_intrusive_call_summary">Show a full screen notification on incoming call even when not on lockscreen</string>
+
+</resources>
diff --git a/res/xml/call_feature_setting.xml b/res/xml/call_feature_setting.xml
index 02034ba..6f30d6d 100644
--- a/res/xml/call_feature_setting.xml
+++ b/res/xml/call_feature_setting.xml
@@ -70,6 +70,12 @@
         android:summary="@string/sum_cdma_call_settings"
         android:persistent="false" />
 
+    <SwitchPreference
+        android:key="use_intrusive_call"
+        android:title="@string/use_intrusive_call_title"
+        android:summary="@string/use_intrusive_call_summary"
+        android:persistent="false" />
+
     <PreferenceScreen
         android:key="ims_settings_key"
         android:title="@string/ims_settings"
diff --git a/src/com/android/phone/CallFeaturesSetting.java b/src/com/android/phone/CallFeaturesSetting.java
index 03034ba..6f30d6d 100644
--- a/src/com/android/phone/CallFeaturesSetting.java
+++ b/src/com/android/phone/CallFeaturesSetting.java
@@ -106,6 +106,8 @@
     private static final String PHONE_ACCOUNT_SETTINGS_KEY =
             "phone_account_settings_preference_screen";
 
+    private static final String USE_INTRUSIVE_CALL_KEY = "use_intrusive_call";
+
     private static final String ENABLE_VIDEO_CALLING_KEY = "button_enable_video_calling";
 
     private Phone mPhone;
@@ -121,6 +121,7 @@
     private PreferenceScreen mVoicemailSettingsScreen;
     private SwitchPreference mEnableVideoCalling;
     private PreferenceScreen mButtonBlacklist;
+    private SwitchPreference mUseIntrusiveCall;
 
     /*
      * Click Listeners, handle click based on objects attached to UI.
@@ -173,6 +173,10 @@ public void onClick(DialogInterface dialog, int which) {
                         .show();
                 return false;
             }
+        } else if (preference == mUseIntrusiveCall) {
+            final boolean val = (Boolean) objValue;
+            Settings.System.putInt(getContentResolver(),
+                    Settings.System.USE_INTRUSIVE_CALL, val ? 1 : 0);
         }
 
         // Always let the preference setting proceed.
@@ -232,6 +232,13 @@ protected void onResume() {
 
         mEnableVideoCalling = (SwitchPreference) findPreference(ENABLE_VIDEO_CALLING_KEY);
 
+        mUseIntrusiveCall = (SwitchPreference) findPreference(USE_INTRUSIVE_CALL_KEY);
+        if (mUseIntrusiveCall != null) {
+            mUseIntrusiveCall.setChecked(Settings.System.getInt(getContentResolver(),
+                Settings.System.USE_INTRUSIVE_CALL, 0) != 0);
+            mUseIntrusiveCall.setOnPreferenceChangeListener(this);
+        }
+
         PersistableBundle carrierConfig =
                 PhoneGlobals.getInstance().getCarrierConfigForSubId(mPhone.getSubId());
 
