diff --git a/src/com/android/settings/deviceinfo/StorageVolumePreference.java b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
index 190f270..2189001 100644
--- a/src/com/android/settings/deviceinfo/StorageVolumePreference.java
+++ b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
@@ -76,7 +76,9 @@ public StorageVolumePreference(Context context, VolumeInfo volume, int color) {
             final String used = Formatter.formatFileSize(context, usedBytes);
             final String total = Formatter.formatFileSize(context, totalBytes);
             setSummary(context.getString(R.string.storage_volume_summary, used, total));
-            mUsedPercent = (int) ((usedBytes * 100) / totalBytes);
+            if (totalBytes > 0) {
+                mUsedPercent = (int) ((usedBytes * 100) / totalBytes);
+            }
 
             if (freeBytes < mStorageManager.getStorageLowBytes(path)) {
                 mColor = context.getColor(R.color.storage_volume_color_warning);
