 Add missing lock discovered by GuardedBy checker.

diff --git a/src/com/android/dialer/voicemail/VoicemailPlaybackLayout.java b/src/com/android/dialer/voicemail/VoicemailPlaybackLayout.java
index 816e1ab..d80ed9f 100644
--- a/src/com/android/dialer/voicemail/VoicemailPlaybackLayout.java
+++ b/src/com/android/dialer/voicemail/VoicemailPlaybackLayout.java
@@ -114,6 +114,7 @@ public void stopUpdating() {
             }
         }
 
+        @GuardedBy("mLock")
         private void cancelPendingRunnables() {
             if (mScheduledFuture != null) {
                 mScheduledFuture.cancel(true);
