diff --git a/Android.mk b/Android.mk
index 7ed9337..1c8b3c0 100644
--- a/Android.mk
+++ b/Android.mk
@@ -20,14 +20,14 @@ common_SRC_FILES := \
 	src/checkAccess.c
 
 common_HOST_FILES := \
-	src/callbacks.c \
-	src/check_context.c \
-	src/freecon.c \
-	src/init.c \
-	src/label.c \
-	src/label_file.c \
-	src/label_android_property.c \
-	src/label_support.c
+	src/host/callbacks.c \
+	src/host/check_context.c \
+	src/host/freecon.c \
+	src/host/init.c \
+	src/host/label.c \
+	src/host/label_file.c \
+	src/host/label_android_property.c \
+	src/host/label_support.c
 
 
 include $(CLEAR_VARS)
