From f7c8c6d16baee379a3682e1b1606ad0fb13ad2d8 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 01/31] set TARGET_GCC_VERSION to 4.9-linaro

---
 core/config.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/config.mk b/core/config.mk
index 5c1a2515f..fe80c3062 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -373,9 +373,9 @@ TARGET_CPU_ABI_LIST_32_BIT := $(subst $(space),$(comma),$(strip $(TARGET_CPU_ABI
 TARGET_CPU_ABI_LIST_64_BIT := $(subst $(space),$(comma),$(strip $(TARGET_CPU_ABI_LIST_64_BIT)))
 
 # GCC version selection
-TARGET_GCC_VERSION := 4.9
+TARGET_GCC_VERSION := 4.9-linaro
 ifdef TARGET_2ND_ARCH
-2ND_TARGET_GCC_VERSION := 4.9
+2ND_TARGET_GCC_VERSION := 4.9-linaro
 endif
 
 # Normalize WITH_STATIC_ANALYZER
-- 
2.11.0

