From 6ef060daa30975e2bd6c609c4e6496dc742c3ba9 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 16/32] Add Board Global CFLAGS

---
 core/combo/select.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/combo/select.mk b/core/combo/select.mk
index eab4c7288..9962d6969 100644
--- a/core/combo/select.mk
+++ b/core/combo/select.mk
@@ -31,6 +31,9 @@ combo_var_prefix := $(combo_2nd_arch_prefix)$(combo_target)
 $(combo_var_prefix)GLOBAL_ARFLAGS := cqsD -format=gnu
 
 $(combo_var_prefix)STATIC_LIB_SUFFIX := .a
+$(combo_var_prefix)GLOBAL_CFLAGS := $(BOARD_GLOBAL_CFLAGS)
+$(combo_var_prefix)RELEASE_CFLAGS := $(BOARD_RELEASE_CFLAGS)
+$(combo_var_prefix)GLOBAL_CPPFLAGS := $(BOARD_GLOBAL_CPPFLAGS)
 
 # Now include the combo for this specific target.
 include $(BUILD_COMBOS)/$(combo_target)$(combo_os_arch).mk
-- 
2.11.0

