From f310a1fda1b5817da3c9fcafa7bf9197c66ad058 Mon Sep 17 00:00:00 2001
From: Andrew Dodd <atd7@cornell.edu>
Date: Sun, 9 Nov 2014 18:06:04 -0500
Subject: [PATCH 22/32] Add busybox to CLANG exclusion list

---
 core/clang/config.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/clang/config.mk b/core/clang/config.mk
index 408f688c6..e09ca55e5 100644
--- a/core/clang/config.mk
+++ b/core/clang/config.mk
@@ -75,6 +75,7 @@ LOCAL_CLANG_EXCEPTION_PROJECTS = \
   device/google/contexthub/ \
   device/huawei/angler/ \
   device/lge/bullhead/ \
+  external/busybox/ \
   external/gentoo/integration/ \
   hardware/qcom/ \
   test/vts/hals/camera/bullhead/ \
-- 
2.11.0

