From 0484c1c497170fce5e67decf833dec88ae4d206d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 23/31] Compressed recovery gzip

---
 core/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index b14d916d6..13bb22357 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1346,6 +1346,8 @@ $(recovery_uncompressed_ramdisk): $(MKBOOTFS) $(ADBD) \
 		$(hide) $(MKBOOTFS) $(TARGET_RECOVERY_ROOT_OUT) > $@
 		@echo ----- Copy uncompressed recovery ------
 		$(hide) cp $(PRODUCT_OUT)/ramdisk-recovery.cpio $(PRODUCT_OUT)/install/bin/recovery.cpio
+		@echo ----- Compressed recovery gzip ------
+		$(hide) gzip $(PRODUCT_OUT)/install/bin/recovery.cpio
 
 $(recovery_ramdisk): $(MINIGZIP) \
 		$(recovery_uncompressed_ramdisk)
-- 
2.11.0

