From 93c6fafba51f6af8620737ea3957dd262425cb9a Mon Sep 17 00:00:00 2001
From: maxwen <max.weninger@gmail.com>
Date: Sat, 9 Dec 2017 03:06:47 +0100
Subject: [PATCH 26/31] build: add TARGET_SPECIFIC_HEADER_PATH to config

---
 core/soong_config.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/soong_config.mk b/core/soong_config.mk
index 1b012fff5..b01fe99d1 100644
--- a/core/soong_config.mk
+++ b/core/soong_config.mk
@@ -88,7 +88,8 @@ $(SOONG_VARIABLES): FORCE lineage_soong
 	echo ''; \
 	echo '    "BtConfigIncludeDir": "$(BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR)",'; \
 	echo ''; \
-	echo '    "DeviceKernelHeaders": $(call json_list,$(strip $(TARGET_PROJECT_SYSTEM_INCLUDES)))'; \
+	echo '    "DeviceKernelHeaders": $(call json_list,$(strip $(TARGET_PROJECT_SYSTEM_INCLUDES))),'; \
+	echo '    "TargetHeaderPath": "$(TARGET_SPECIFIC_HEADER_PATH)"'; \
 	echo '}') >> $(SOONG_VARIABLES_TMP); \
 	if ! cmp -s $(SOONG_VARIABLES_TMP) $(SOONG_VARIABLES); then \
 	  mv $(SOONG_VARIABLES_TMP) $(SOONG_VARIABLES); \
-- 
2.11.0

