From 04ce02f7706ff7b28d6174bb78138de2828cdfb4 Mon Sep 17 00:00:00 2001
From: Tobias Tefke <tobias.tefke@gmail.com>
Date: Sat, 3 Mar 2018 13:57:36 +0100
Subject: [PATCH 27/31] Fix config Android Go

---
 target/product/go_defaults_common.mk | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/target/product/go_defaults_common.mk b/target/product/go_defaults_common.mk
index fbb95677f..1773b5778 100644
--- a/target/product/go_defaults_common.mk
+++ b/target/product/go_defaults_common.mk
@@ -18,7 +18,6 @@
 
 # Set lowram options
 PRODUCT_PROPERTY_OVERRIDES += \
-     ro.config.low_ram=true \
      ro.lmk.critical_upgrade=true \
      ro.lmk.upgrade_pressure=40
 
@@ -50,8 +49,3 @@ PRODUCT_DEX_PREOPT_BOOT_IMAGE_PROFILE_LOCATION := frameworks/base/config/boot-im
 PRODUCT_PROPERTY_OVERRIDES += \
      pm.dexopt.shared=quicken
 
-# Default heap sizes. Allow up to 256m for large heaps to make sure a single app
-# doesn't take all of the RAM.
-PRODUCT_PROPERTY_OVERRIDES += dalvik.vm.heapgrowthlimit=128m
-PRODUCT_PROPERTY_OVERRIDES += dalvik.vm.heapsize=256m
-
-- 
2.11.0

