From de437a2de83f46a2a109b3917a826d1de305f3bb Mon Sep 17 00:00:00 2001
From: Dan Pasanen <invisiblek@cyanogenmod.org>
Date: Mon, 5 Sep 2016 18:17:44 -0500
Subject: [PATCH 28/31] build: don't dex preopt by default on linux

---
 core/dex_preopt.mk | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/core/dex_preopt.mk b/core/dex_preopt.mk
index 69ab37a99..692fb08d9 100644
--- a/core/dex_preopt.mk
+++ b/core/dex_preopt.mk
@@ -28,18 +28,18 @@ install-on-system-other = $(filter-out $(PRODUCT_DEXPREOPT_SPEED_APPS) $(PRODUCT
 
 # The default values for pre-opting: always preopt PIC.
 # Conditional to building on linux, as dex2oat currently does not work on darwin.
-ifeq ($(HOST_OS),linux)
-  WITH_DEXPREOPT ?= true
+#ifeq ($(HOST_OS),linux)
+#  WITH_DEXPREOPT ?= true
 # For an eng build only pre-opt the boot image and system server. This gives reasonable performance
 # and still allows a simple workflow: building in frameworks/base and syncing.
-  ifneq (user,$(TARGET_BUILD_VARIANT))
-    WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY ?= true
-  endif
+#  ifeq (eng,$(TARGET_BUILD_VARIANT))
+#    WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY ?= true
+#  endif
 # Add mini-debug-info to the boot classpath unless explicitly asked not to.
-  ifneq (false,$(WITH_DEXPREOPT_DEBUG_INFO))
-    PRODUCT_DEX_PREOPT_BOOT_FLAGS += --generate-mini-debug-info
-  endif
-endif
+#  ifneq (false,$(WITH_DEXPREOPT_DEBUG_INFO))
+#    PRODUCT_DEX_PREOPT_BOOT_FLAGS += --generate-mini-debug-info
+#  endif
+#endif
 
 GLOBAL_DEXPREOPT_FLAGS :=
 
-- 
2.11.0

