From 273067bc174efe5eaf799a902e846dc3a0383448 Mon Sep 17 00:00:00 2001
From: Andrew Dodd <atd7@cornell.edu>
Date: Sun, 9 Nov 2014 18:06:04 -0500
Subject: [PATCH] Add busybox to CLANG exclusion list

diff --git a/core/clang/config.mk b/core/clang/config.mk
index 0e59930..a3154f2 100644
--- a/core/clang/config.mk
+++ b/core/clang/config.mk
@@ -74,6 +74,7 @@
   bionic/tests/ \
   device/huawei/angler/ \
   device/lge/bullhead/ \
+  external/busybox/ \
   external/gentoo/integration/ \
   hardware/qcom/ \
   test/vts/hals/camera/bullhead/ \
