From de84b4c160a6c221b0d7626791259197023f0074 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 01/40] Add init.kernel.rc

---
 rootdir/init.rc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index d6e068a91..96f102587 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -10,6 +10,7 @@ import /init.${ro.hardware}.rc
 import /vendor/etc/init/hw/init.${ro.hardware}.rc
 import /init.usb.configfs.rc
 import /init.${ro.zygote}.rc
+import /init.kernel.rc
 
 on early-init
     # Set init and its forked children's oom_adj.
-- 
2.11.0

