From eaaeaecb520940006da3ea1cbe4b7dba0883ce2c Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 03/16] Remove flash recovery service

---
 rootdir/init.rc | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 02337116c..c3e47d323 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -765,12 +765,3 @@ on property:ro.debuggable=1
     chmod 0775 /data/misc/wmtrace
     start console
 
-service flash_recovery /system/bin/install-recovery.sh
-    class main
-    oneshot
-    disabled
-
-# update recovery if enabled
-on property:persist.sys.recovery_update=true
-    start flash_recovery
-
-- 
2.11.0

