From 0093081ebfb0b1a563311df1944d8bdaf3bd834a Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 09/40] STE Flattenable fix

---
 libutils/include/utils/Flattenable.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libutils/include/utils/Flattenable.h b/libutils/include/utils/Flattenable.h
index 3f64db593..45f4031bb 100644
--- a/libutils/include/utils/Flattenable.h
+++ b/libutils/include/utils/Flattenable.h
@@ -63,8 +63,10 @@ public:
     // write a POD structure
     template<typename T>
     static void write(void*& buffer, size_t& size, const T& value) {
+/*
         static_assert(std::is_trivially_copyable<T>::value,
                       "Cannot flatten a non-trivially-copyable type");
+*/
         memcpy(buffer, &value, sizeof(T));
         advance(buffer, size, sizeof(T));
     }
@@ -72,8 +74,10 @@ public:
     // read a POD structure
     template<typename T>
     static void read(void const*& buffer, size_t& size, T& value) {
+/*
         static_assert(std::is_trivially_copyable<T>::value,
                       "Cannot unflatten a non-trivially-copyable type");
+*/
         memcpy(&value, buffer, sizeof(T));
         advance(buffer, size, sizeof(T));
     }
-- 
2.11.0

