From 57cf7d54e28454277d446aa9ccdb4f1a0092b481 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Fri, 6 Jan 2017 13:53:13 +0700
Subject: [PATCH 10/29] system-core_W6-adapt-to-our-sdcardfs.patch

Change-Id: I597f29a29c798b62f3f3626b3d061481f1b4a95c
---
 sdcard/sdcard.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sdcard/sdcard.c b/sdcard/sdcard.c
index 281d23e..094d59e 100644
--- a/sdcard/sdcard.c
+++ b/sdcard/sdcard.c
@@ -1964,8 +1964,7 @@ static int sdcardfs_setup(const char *source_path, const char *dest_path, uid_t
     char opts[256];
 
     snprintf(opts, sizeof(opts),
-            "fsuid=%d,fsgid=%d,%smask=%d,userid=%d,gid=%d",
-            fsuid, fsgid, multi_user?"multiuser,":"", mask, userid, gid);
+            "uid=1023,gid=1023,wgid=1023,derive=unified");
 
     if (mount(source_path, dest_path, "sdcardfs",
                         MS_NOSUID | MS_NODEV | MS_NOEXEC | MS_NOATIME, opts) != 0) {
-- 
2.9.3

