From 2b1356b2c25005244e7b2f58f5c7e35f70d8e06a Mon Sep 17 00:00:00 2001
From: xyyx <xyyx@mail.ru>
Date: Tue, 7 Aug 2018 08:38:17 +0800
Subject: [PATCH 12/40] Allow run services without selinux domain

---
 init/service.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/init/service.cpp b/init/service.cpp
index 7c50165c1..be7d2b141 100644
--- a/init/service.cpp
+++ b/init/service.cpp
@@ -83,12 +83,6 @@ static std::string ComputeContextFromExecutable(std::string& service_name,
         computed_context = new_con;
         free(new_con);
     }
-    if (rc == 0 && computed_context == mycon.get()) {
-        LOG(ERROR) << "service " << service_name << " does not have a SELinux domain defined";
-        if (selinux_status_getenforce() > 0) {
-            return "";
-        }
-    }
     if (rc < 0) {
         LOG(ERROR) << "could not get context while starting '" << service_name << "'";
         return "";
-- 
2.11.0

