From 9c574fe973281166818ca6461cf5cec2cc2a1898 Mon Sep 17 00:00:00 2001
From: beanstown106 <nbenis106@gmail.com>
Date: Fri, 9 Oct 2015 00:08:41 -0400
Subject: [PATCH 16/40] set permissions on /system/addon.d/ and files within.

---
 libcutils/fs_config.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libcutils/fs_config.cpp b/libcutils/fs_config.cpp
index cb15b57b8..bfbf77179 100644
--- a/libcutils/fs_config.cpp
+++ b/libcutils/fs_config.cpp
@@ -84,6 +84,7 @@ static const struct fs_path_config android_dirs[] = {
     { 00750, AID_ROOT,         AID_SHELL,        0, "sbin" },
     { 00777, AID_ROOT,         AID_ROOT,         0, "sdcard" },
     { 00751, AID_ROOT,         AID_SDCARD_R,     0, "storage" },
+    { 00755, AID_ROOT,         AID_ROOT,         0, "system/addon.d" },
     { 00755, AID_ROOT,         AID_SHELL,        0, "system/bin" },
     { 00755, AID_ROOT,         AID_ROOT,         0, "system/etc/ppp" },
     { 00755, AID_ROOT,         AID_SHELL,        0, "system/vendor" },
@@ -125,6 +126,7 @@ static const char* conf[][2] = {
 
 static const struct fs_path_config android_files[] = {
     // clang-format off
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/addon.d/*" },
     { 00644, AID_SYSTEM,    AID_SYSTEM,    0, "data/app/*" },
     { 00644, AID_SYSTEM,    AID_SYSTEM,    0, "data/app-ephemeral/*" },
     { 00644, AID_SYSTEM,    AID_SYSTEM,    0, "data/app-private/*" },
-- 
2.11.0

