From 4f51d8155d550d7af2989ab522a83e6f9f1f67a0 Mon Sep 17 00:00:00 2001
From: Dan Pasanen <invisiblek@cyanogenmod.org>
Date: Thu, 10 Nov 2016 21:24:16 -0600
Subject: [PATCH 22/40] init.rc: set DOWNLOAD_CACHE based on prop

---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index fe6915e6c..2da7b28e8 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -237,7 +237,7 @@ on init
     # expecting it to point to /proc/self/fd
     symlink /proc/self/fd /dev/fd
 
-    export DOWNLOAD_CACHE /data/cache
+    export DOWNLOAD_CACHE /cache
 
     # set RLIMIT_NICE to allow priorities from 19 to -20
     setrlimit 13 40 40
-- 
2.11.0

