sdcard: adapt to our sdcardfs implementation 

diff --git a/sdcard/sdcard.c b/sdcard/sdcard.c
index 04a25f0..a0397a0 100644
--- a/sdcard/sdcard.c
+++ b/sdcard/sdcard.c
@@ -1792,10 +1792,7 @@ static int fuse_setup(struct fuse* fuse, gid_t gid, mode_t mask, bool use_sdcard
 
     if (use_sdcardfs) {
         snprintf(opts, sizeof(opts),
-                "%sfsuid=%u,fsgid=%u,userid=%d,gid=%u,mask=%u,reserved_mb=20",
-                (fuse->global->multi_user ? "multiuser," : ""),
-                fuse->global->uid, fuse->global->gid,
-                fuse->global->root.userid, gid, mask);
+                "uid=1023,gid=1023,wgid=1023,derive=unified");
         if (mount(fuse->global->source_path, fuse->dest_path, "sdcardfs",
                     MS_NOSUID | MS_NODEV | MS_NOEXEC | MS_NOATIME, opts) != 0) {
             ERROR("failed to mount sdcardfs filesystem: %s\n",  strerror(errno));
