From c2cd12684fd7f58e8f6f328e0dbf6ccf887391c1 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm@cyngn.com>
Date: Mon, 24 Nov 2014 04:38:08 -0800
Subject: [PATCH] healthd: Use project pathmap for recovery

diff --git a/healthd/Android.mk b/healthd/Android.mk
index 8b59964..0e685a7 100644
--- a/healthd/Android.mk
+++ b/healthd/Android.mk
@@ -70,7 +70,7 @@
     AnimationParser.cpp
 
 LOCAL_MODULE := libhealthd_charger
-LOCAL_C_INCLUDES := bootable/recovery $(LOCAL_PATH)/include
+LOCAL_C_INCLUDES := $(call project-path-for,recovery) $(LOCAL_PATH)/include
 LOCAL_EXPORT_C_INCLUDE_DIRS := \
     $(LOCAL_PATH) \
     $(LOCAL_PATH)/include
