From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] introduce lpm boot offline charging
diff --git a/init/Android.mk b/init/Android.mk
index f3bb7d6..a5e0d0e 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -61,7 +61,10 @@ endif
 
 SYSTEM_CORE_INIT_DEFINES := BOARD_CHARGING_MODE_BOOTING_LPM \
     BOARD_CHARGING_CMDLINE_NAME \
-    BOARD_CHARGING_CMDLINE_VALUE
+    BOARD_CHARGING_CMDLINE_VALUE \
+    BOARD_LPM_BOOT_ARGUMENT_NAME \
+    BOARD_LPM_BOOT_ARGUMENT_VALUE
+
 
 $(foreach system_core_init_define,$(SYSTEM_CORE_INIT_DEFINES), \
   $(if $($(system_core_init_define)), \
diff --git a/init/init.cpp b/init/init.cpp
index a9577af..79726e7 100755
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -88,6 +88,11 @@
 #define BOARD_CHARGING_CMDLINE_VALUE "true"
 #endif
 
+#ifndef BOARD_LPM_BOOT_ARGUMENT_NAME
+#define BOARD_LPM_BOOT_ARGUMENT_NAME "lpm_boot"
+#define BOARD_LPM_BOOT_ARGUMENT_VALUE "1"
+#endif
+
 static char qemu[32];
 static char battchg_pause[32];
 
@@ -96,6 +101,8 @@ static time_t process_needs_restart;
 
 const char *ENV[32];
 
+static unsigned lpm_bootmode = 0;
+
 static int epoll_fd = -1;
 
 static std::unique_ptr<Timer> waiting_for_prop(nullptr);
@@ -488,5 +495,11 @@
     if (key == "qemu") {
         strlcpy(qemu, value.c_str(), sizeof(qemu));
+#ifdef BOARD_LPM_BOOT_ARGUMENT_NAME
+    } else if (key == BOARD_LPM_BOOT_ARGUMENT_NAME) {
+        if (value == BOARD_LPM_BOOT_ARGUMENT_VALUE) {
+            lpm_bootmode = 1;
+        }
+#endif
     } else if (key == BOARD_CHARGING_CMDLINE_NAME) {
         strlcpy(battchg_pause, value.c_str(), sizeof(battchg_pause));
     } else if (android::base::StartsWith(key, "androidboot.")) {
@@ -995,7 +1008,7 @@
 
 static int charging_mode_booting(void) {
 #ifndef BOARD_CHARGING_MODE_BOOTING_LPM
-    return 0;
+    return lpm_bootmode;
 #else
     int f;
     char cmb;
