From be5fca21f85ce707678b47227a8b0d1877bd686c Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 24/57] STE AV Fix [2]

---
 media/ndk/NdkImageReader.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/ndk/NdkImageReader.cpp b/media/ndk/NdkImageReader.cpp
index fd6ecb51d..a6792ba01 100644
--- a/media/ndk/NdkImageReader.cpp
+++ b/media/ndk/NdkImageReader.cpp
@@ -149,7 +149,9 @@ AImageReader::BufferRemovedListener::onBufferFreed(const wp<GraphicBuffer>& grap
     msg->setPointer(
         AImageReader::kCallbackFpKey, (void*) mListener.onBufferRemoved);
     msg->setPointer(AImageReader::kContextKey, mListener.context);
+#ifndef STE_HARDWARE
     msg->setObject(AImageReader::kGraphicBufferKey, gBuffer);
+#endif
     msg->post();
 }
 
-- 
2.11.0

