From 5202c85fedb0ab8f86d1ae4249c9381fc555f4a7 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 41/57] Add debug ACodec to allow using Treble

---
 media/libstagefright/ACodec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 13d3a9040..4d733845d 100755
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -576,7 +576,7 @@ ACodec::ACodec()
 
     changeState(mUninitializedState);
 
-    mTrebleFlag = false;
+    mTrebleFlag = bool(property_get_bool("debug.treble_omx", false));
 }
 
 ACodec::~ACodec() {
-- 
2.11.0

