From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] av: Revert mGraphicBufferEnabled

diff --git a/media/libstagefright/omx/include/media/stagefright/omx/OMXNodeInstance.h b/media/libstagefright/omx/include/media/stagefright/omx/OMXNodeInstance.h
index 7a7bee7..f7ff0b7 100644
--- a/media/libstagefright/omx/include/media/stagefright/omx/OMXNodeInstance.h
+++ b/media/libstagefright/omx/include/media/stagefright/omx/OMXNodeInstance.h
@@ -160,7 +160,6 @@
         kSecureBufferTypeNativeHandle,
     };
     SecureBufferType mSecureBufferType[2];
-    bool mGraphicBufferEnabled[2];
 
     // Following are OMX parameters managed by us (instead of the component)
     // OMX_IndexParamMaxFrameDurationForBitrateControl
diff --git a/media/libstagefright/omx/OMXNodeInstance.cpp b/media/libstagefright/omx/OMXNodeInstance.cpp
index cf7f62d..f49c734 100644
--- a/media/libstagefright/omx/OMXNodeInstance.cpp
+++ b/media/libstagefright/omx/OMXNodeInstance.cpp
@@ -372,8 +372,6 @@
     mPortMode[1] = IOMX::kPortModePresetByteBuffer;
     mSecureBufferType[0] = kSecureBufferTypeUnknown;
     mSecureBufferType[1] = kSecureBufferTypeUnknown;
-    mGraphicBufferEnabled[0] = false;
-    mGraphicBufferEnabled[1] = false;
     mIsSecure = AString(name).endsWith(".secure");
     mLegacyAdaptiveExperiment = ADebug::isExperimentEnabled("legacy-adaptive");
 }
@@ -818,12 +816,6 @@
             } else if (mSecureBufferType[portIndex] == kSecureBufferTypeUnknown) {
                 mSecureBufferType[portIndex] = kSecureBufferTypeOpaque;
             }
-        } else {
-            if (err == OMX_ErrorNone) {
-                mGraphicBufferEnabled[portIndex] = enable;
-            } else if (enable) {
-                mGraphicBufferEnabled[portIndex] = false;
-            }
         }
     } else {
         CLOG_ERROR_IF(enable, getExtensionIndex, err, "%s", name);
@@ -1092,12 +1084,6 @@
     OMX_ERRORTYPE err = OMX_ErrorNone;
     bool isMetadata = mMetadataType[portIndex] != kMetadataBufferTypeInvalid;
 
-    if (!isMetadata && mGraphicBufferEnabled[portIndex]) {
-        ALOGE("b/62948670");
-        android_errorWriteLog(0x534e4554, "62948670");
-        return INVALID_OPERATION;
-    }
-
     size_t paramsSize;
     void* paramsPointer;
     if (params != NULL && hParams != NULL) {
@@ -1283,13 +1269,6 @@
                 portIndex, graphicBuffer, buffer);
     }
 
-    if (!mGraphicBufferEnabled[portIndex]) {
-        // Report error if this is not in graphic buffer mode.
-        ALOGE("b/62948670");
-        android_errorWriteLog(0x534e4554, "62948670");
-        return INVALID_OPERATION;
-    }
-
     // See if the newer version of the extension is present.
     OMX_INDEXTYPE index;
 #ifndef STE_HARDWARE
