From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Nuplayer: Ensure AudioSink is closed before clearing Renderer

Close AudioSink object explicitly before clearing NuPlayerRenderer
to avoid race conditions between NuPlayerRenderer and AudioSink callback
thread.

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index 083421f..ed7a200 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -1308,6 +1308,9 @@ void NuPlayer::onMessageReceived(const sp<AMessage> &msg) {
                         FLUSH_CMD_SHUTDOWN /* audio */,
                         FLUSH_CMD_SHUTDOWN /* video */));
 
+            mDeferredActions.push_back(
+                    new SimpleAction(&NuPlayer::closeAudioSink));
+
             mDeferredActions.push_back(
                     new SimpleAction(&NuPlayer::performReset));
 
