diff --git a/hostapd/Android.mk b/hostapd/Android.mk
index 200f300..bb89d10 100644
--- a/hostapd/Android.mk
+++ b/hostapd/Android.mk
@@ -20,6 +20,7 @@ include $(LOCAL_PATH)/android.config
 L_CFLAGS = -DWPA_IGNORE_CONFIG_ERRORS
 
 L_CFLAGS += -DVERSION_STR_POSTFIX=\"-$(PLATFORM_VERSION)\"
+L_CFLAGS += -Os
 
 # Set Android log name
 L_CFLAGS += -DANDROID_LOG_NAME=\"hostapd\"
diff --git a/hs20/client/Android.mk b/hs20/client/Android.mk
index 200f310..bb89d10 100644
--- a/hs20/client/Android.mk
+++ b/hs20/client/Android.mk
@@ -30,6 +30,8 @@ L_CFLAGS += -DCONFIG_CTRL_IFACE
 L_CFLAGS += -DCONFIG_CTRL_IFACE_UNIX
 L_CFLAGS += -DCONFIG_CTRL_IFACE_CLIENT_DIR=\"/data/misc/wifi/sockets\"
 
+L_CFLAGS += -Os
+
 OBJS = spp_client.c
 OBJS += oma_dm_client.c
 OBJS += osu_client.c
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 200f320..bb89d10 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -18,6 +18,7 @@ include $(LOCAL_PATH)/android.config
 L_CFLAGS = -DWPA_IGNORE_CONFIG_ERRORS
 
 L_CFLAGS += -DVERSION_STR_POSTFIX=\"-$(PLATFORM_VERSION)\"
+L_CFLAGS += -Os
 
 # Set Android log name
 L_CFLAGS += -DANDROID_LOG_NAME=\"wpa_supplicant\"
