From 93514b977fe7da61d4494cf0df7414b7f1b69242 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sat, 15 Oct 2016 18:57:38 +0300
Subject: [PATCH 11/28] Adjustments for M-compatible sensors ABI

---
 core/jni/android_hardware_SensorManager.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/jni/android_hardware_SensorManager.cpp b/core/jni/android_hardware_SensorManager.cpp
index 5b327d40ac6..dae96523f88 100644
--- a/core/jni/android_hardware_SensorManager.cpp
+++ b/core/jni/android_hardware_SensorManager.cpp
@@ -201,7 +201,11 @@ nativeGetSensorAtIndex(JNIEnv *env, jclass clazz, jlong sensorManager, jobject s
     SensorManager* mgr = reinterpret_cast<SensorManager*>(sensorManager);
 
     Sensor const* const* sensorList;
+#ifdef COMPAT_SENSORS_M
+    const ssize_t count = mgr->getSensorList(&sensorList);
+#else
     ssize_t count = mgr->getSensorList(&sensorList);
+#endif
     if (ssize_t(index) >= count) {
         return false;
     }
-- 
2.11.0

