From 739a92d17ee6b674be4dbd25bda1dc54d2c1c17d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 169/296] Add NewSettings proto

---
 proto/src/metrics_constants.proto | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/proto/src/metrics_constants.proto b/proto/src/metrics_constants.proto
index a27515c0266..91e6a5a2830 100644
--- a/proto/src/metrics_constants.proto
+++ b/proto/src/metrics_constants.proto
@@ -4604,5 +4604,8 @@ message MetricsEvent {
 
     // Add new aosp constants above this line.
     // END OF AOSP CONSTANTS
+
+    // NewSettings constant
+    NEW_SETTINGS = 1500;
   }
 }
-- 
2.11.0

