From 103cec1b5682d4cf9c0a5f93a4a9ae85b06949d4 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 255/296] SelectionActionModeHelper: Prevent NPE

---
 core/java/android/widget/SelectionActionModeHelper.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/java/android/widget/SelectionActionModeHelper.java b/core/java/android/widget/SelectionActionModeHelper.java
index 384e254e205..29e033689fc 100644
--- a/core/java/android/widget/SelectionActionModeHelper.java
+++ b/core/java/android/widget/SelectionActionModeHelper.java
@@ -446,6 +446,11 @@ final class SelectionActionModeHelper {
                 @SelectionEvent.ActionType int action,
                 @Nullable TextClassification classification) {
             try {
+                if (mText == null || mDelegate == null) {
+                    Log.d(LOG_TAG, "logSelectionAction not logged.");
+                    return;
+                }
+
                 Preconditions.checkArgumentInRange(start, 0, mText.length(), "start");
                 Preconditions.checkArgumentInRange(end, start, mText.length(), "end");
                 int[] wordIndices = getWordDelta(start, end);
-- 
2.11.0

