From d96711bf34642260d4684282eab2381c0829d3f2 Mon Sep 17 00:00:00 2001
From: Akhil Narang <akhilnarang.1999@gmail.com>
Date: Tue, 26 Jun 2018 16:55:14 +0530
Subject: [PATCH 288/296] Toast: Fix duplicated to add View of Toast

---
 core/java/android/view/WindowManagerGlobal.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/java/android/view/WindowManagerGlobal.java b/core/java/android/view/WindowManagerGlobal.java
index 6d3c5ecb2ea..c1c5cbee839 100644
--- a/core/java/android/view/WindowManagerGlobal.java
+++ b/core/java/android/view/WindowManagerGlobal.java
@@ -356,6 +356,7 @@ public final class WindowManagerGlobal {
                 root.setView(view, wparams, panelParentView);
             } catch (RuntimeException e) {
                 // BadTokenException or InvalidDisplayException, clean up.
+                index = findViewLocked(view, false);
                 if (index >= 0) {
                     removeViewLocked(index, true);
                 }
-- 
2.11.0

