From 909073ee25ba5fd9aa52b724c314bbee4e90a322 Mon Sep 17 00:00:00 2001
From: Toha <tohenk@yahoo.com>
Date: Sat, 10 Mar 2018 09:18:29 +0700
Subject: [PATCH] TelephonyManager: Prevent NPE when registering phone state listener

diff --git a/telephony/java/android/telephony/TelephonyManager.java b/telephony/java/android/telephony/TelephonyManager.java
index 011d76b0a54..1e0a1d281ed 100644
--- a/telephony/java/android/telephony/TelephonyManager.java
+++ b/telephony/java/android/telephony/TelephonyManager.java
@@ -3552,7 +3552,7 @@ private ITelephonyRegistry getTelephonyRegistry() {
      *               LISTEN_ flags.
      */
     public void listen(PhoneStateListener listener, int events) {
-        if (mContext == null) return;
+        if (mContext == null || listener == null) return;
         try {
             boolean notifyNow = (getITelephony() != null);
             // If the listener has not explicitly set the subId (for example, created with the
