From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] MTP Dirty Hack (use mtp as default after connect to usb) (1/3)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 0132140..006f681 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -3960,6 +3960,12 @@ public boolean validate(String value) {
          public static final String VOLUME_KEY_CURSOR_CONTROL = "volume_key_cursor_control";
 
         /**
+         * Whether to use the MTP by default after connecting to PC
+         * @hide
+         */
+        public static final String MTP_DIRTY_HACK = "mtp_dirty_hack";
+
+        /**
          * Settings to backup. This is here so that it's in the same place as the settings
          * keys and easy to update.
          *
diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 0142140..006f683 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -1087,6 +1087,14 @@ private void updateUsbNotification(boolean force) {
                 titleRes = com.android.internal.R.string.usb_unsupported_audio_accessory_title;
                 id = SystemMessage.NOTE_USB_AUDIO_ACCESSORY_NOT_SUPPORTED;
             } else if (mConnected) {
+
+                if (Settings.System.getInt(mContext.getContentResolver(),
+                        Settings.System.MTP_DIRTY_HACK, 1) == 1
+                        && ("".equals(SystemProperties.get("mtp_hack")))) {
+                    mUsbDataUnlocked = true;
+                    setCurrentFunctions(UsbManager.USB_FUNCTION_MTP, mUsbDataUnlocked);
+                }
+
                 if (!mUsbDataUnlocked) {
                     if (mSourcePower) {
                         titleRes = com.android.internal.R.string.usb_supplying_notification_title;
diff --git a/services/core/java/com/android/server/connectivity/Tethering.java b/services/core/java/com/android/server/connectivity/Tethering.java
index 0142146..006f689 100644
--- a/services/core/java/com/android/server/connectivity/Tethering.java
+++ b/services/core/java/com/android/server/connectivity/Tethering.java
@@ -70,6 +70,7 @@
 import android.os.Parcel;
 import android.os.RemoteException;
 import android.os.ResultReceiver;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.provider.Settings;
 import android.telephony.CarrierConfigManager;
@@ -1035,6 +1036,7 @@ public int setUsbTethering(boolean enable) {
 
         synchronized (mPublicSync) {
             if (enable) {
+                SystemProperties.set("mtp_hack", "true");
                 if (mRndisEnabled) {
                     final long ident = Binder.clearCallingIdentity();
                     try {
@@ -1048,6 +1050,7 @@ public int setUsbTethering(boolean enable) {
                     usbManager.setCurrentFunction(UsbManager.USB_FUNCTION_RNDIS, false);
                 }
             } else {
+                SystemProperties.set("mtp_hack", "");
                 final long ident = Binder.clearCallingIdentity();
                 try {
                     tetherMatchingInterfaces(IControlsTethering.STATE_AVAILABLE,
