From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] FileUtils: Avoid null pointer dereference

diff --git a/core/java/android/os/FileUtils.java b/core/java/android/os/FileUtils.java
index 56d6e0a62f94d..4b7e07761e8a1 100644
--- a/core/java/android/os/FileUtils.java
+++ b/core/java/android/os/FileUtils.java
@@ -436,6 +436,7 @@ public static boolean contains(File dir, File file) {
     }
 
     public static boolean contains(String dirPath, String filePath) {
+        if (dirPath == null || filePath == null) return false;
         if (dirPath.equals(filePath)) {
             return true;
         }
