From 89281d97741278ecdb3f011d22ad37226f7d0e9b Mon Sep 17 00:00:00 2001
From: Lei Peng <lei.peng@spreadtrum.com>
Date: Fri, 10 Mar 2017 15:00:02 +0800
Subject: [PATCH] Fix a null pointer exception in FragmentManager

diff --git a/core/java/android/app/FragmentManager.java b/core/java/android/app/FragmentManager.java
index bfaf332..6cb4cc5 100644
--- a/core/java/android/app/FragmentManager.java
+++ b/core/java/android/app/FragmentManager.java
@@ -1349,7 +1349,7 @@
                         if (f.mView != null) {
                             // Need to save the current view state if not
                             // done already.
-                            if (mHost.onShouldSaveFragmentState(f) && f.mSavedViewState == null) {
+                            if (mHost != null && mHost.onShouldSaveFragmentState(f) && f.mSavedViewState == null) {
                                 saveFragmentViewState(f);
                             }
                         }
