From d912374400aa3a3e83d3a13c266fd1d7ddcec418 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sat, 15 Oct 2016 18:57:38 +0300
Subject: [PATCH] Adjustments for M-compatible sensors ABI

diff --git a/core/jni/android_hardware_SensorManager.cpp b/core/jni/android_hardware_SensorManager.cpp
index 50f23bf..f29bec8 100644
--- a/core/jni/android_hardware_SensorManager.cpp
+++ b/core/jni/android_hardware_SensorManager.cpp
@@ -201,7 +201,11 @@
     SensorManager* mgr = reinterpret_cast<SensorManager*>(sensorManager);
 
     Sensor const* const* sensorList;
+#ifdef COMPAT_SENSORS_M
+    const ssize_t count = mgr->getSensorList(&sensorList);
+#else
     ssize_t count = mgr->getSensorList(&sensorList);
+#endif
     if (ssize_t(index) >= count) {
         return false;
     }
