From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] PowerManager: do not light up HW-buttons when pressing home-key

diff --git a/services/core/jni/com_android_server_power_PowerManagerService.cpp b/services/core/jni/com_android_server_power_PowerManagerService.cpp
index b882986e783..b2c4f35dcb8 100644
--- a/services/core/jni/com_android_server_power_PowerManagerService.cpp
+++ b/services/core/jni/com_android_server_power_PowerManagerService.cpp
@@ -159,7 +159,7 @@ void android_server_PowerManagerService_userActivity(nsecs_t eventTime, int32_t
         JNIEnv* env = AndroidRuntime::getJNIEnv();
 
         int flags = 0;
-        if (keyCode == AKEYCODE_VOLUME_UP || keyCode == AKEYCODE_VOLUME_DOWN) {
+        if (keyCode == AKEYCODE_VOLUME_UP || keyCode == AKEYCODE_VOLUME_DOWN || keyCode == AKEYCODE_HOME) {
             flags |= USER_ACTIVITY_FLAG_NO_BUTTON_LIGHTS;
         }
 
