From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Check wifi feature before starting wifi service

Otherwise, Wifi service could crash SystemServer,for example, emulator.

diff --git a/services/java/com/android/server/SystemServer.java b/services/java/com/android/server/SystemServer.java
index 50340145a2d..2f901c13846 100644
--- a/services/java/com/android/server/SystemServer.java
+++ b/services/java/com/android/server/SystemServer.java
@@ -1090,35 +1090,38 @@ private void startOtherServices() {
                 }
                 traceEnd();
 
-                // Wifi Service must be started first for wifi-related services.
-                traceBeginAndSlog("StartWifi");
-                mSystemServiceManager.startService(WIFI_SERVICE_CLASS);
-                traceEnd();
-                traceBeginAndSlog("StartWifiScanning");
-                mSystemServiceManager.startService(
-                        "com.android.server.wifi.scanner.WifiScanningService");
-                traceEnd();
-
-                if (!disableRtt) {
-                    traceBeginAndSlog("StartWifiRtt");
-                    mSystemServiceManager.startService("com.android.server.wifi.RttService");
-                    traceEnd();
-                }
-
                 if (context.getPackageManager().hasSystemFeature(
-                        PackageManager.FEATURE_WIFI_AWARE)) {
-                    traceBeginAndSlog("StartWifiAware");
-                    mSystemServiceManager.startService(WIFI_AWARE_SERVICE_CLASS);
+                        PackageManager.FEATURE_WIFI)) {
+                    // Wifi Service must be started first for wifi-related services.
+                    traceBeginAndSlog("StartWifi");
+                    mSystemServiceManager.startService(WIFI_SERVICE_CLASS);
                     traceEnd();
-                } else {
-                    Slog.i(TAG, "No Wi-Fi Aware Service (Aware support Not Present)");
-                }
-
-                if (context.getPackageManager().hasSystemFeature(
-                        PackageManager.FEATURE_WIFI_DIRECT)) {
-                    traceBeginAndSlog("StartWifiP2P");
-                    mSystemServiceManager.startService(WIFI_P2P_SERVICE_CLASS);
+                    traceBeginAndSlog("StartWifiScanning");
+                    mSystemServiceManager.startService(
+                            "com.android.server.wifi.scanner.WifiScanningService");
                     traceEnd();
+
+                    if (!disableRtt) {
+                        traceBeginAndSlog("StartWifiRtt");
+                        mSystemServiceManager.startService("com.android.server.wifi.RttService");
+                        traceEnd();
+                    }
+
+                    if (context.getPackageManager().hasSystemFeature(
+                            PackageManager.FEATURE_WIFI_AWARE)) {
+                        traceBeginAndSlog("StartWifiAware");
+                        mSystemServiceManager.startService(WIFI_AWARE_SERVICE_CLASS);
+                        traceEnd();
+                    } else {
+                        Slog.i(TAG, "No Wi-Fi Aware Service (Aware support Not Present)");
+                    }
+
+                    if (context.getPackageManager().hasSystemFeature(
+                            PackageManager.FEATURE_WIFI_DIRECT)) {
+                        traceBeginAndSlog("StartWifiP2P");
+                        mSystemServiceManager.startService(WIFI_P2P_SERVICE_CLASS);
+                        traceEnd();
+                    }
                 }
 
                 if (context.getPackageManager().hasSystemFeature(
