From 4fcb38dbe014ee3502f5fd023cad9a03004457de Mon Sep 17 00:00:00 2001
From: Alex Cruz <mazdarider23@gmail.com>
Date: Fri, 12 Feb 2016 23:26:33 -0500
Subject: [PATCH] Speed up the volume panel timeout

diff --git a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
index 66fa06f..0114644 100644
--- a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
+++ b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
@@ -518,12 +518,12 @@
         if (mAccessibility.mFeedbackEnabled) return 20000;
         if (mHovering) return 16000;
         if (mSafetyWarning != null) return 5000;
-        if (mExpanded || mExpandButtonAnimationRunning) return 5000;
+        if (mExpanded || mExpandButtonAnimationRunning) return 1500;
         if (mActiveStream == AudioManager.STREAM_MUSIC) return 1500;
         if (mZenFooter.shouldShowIntroduction()) {
-            return 6000;
+            return 6000;
         }
-        return 3000;
+        return 1500;
     }
 
     protected void dismissH(int reason) {
