From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Fix binderDied NetworkRequest

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index e184288..496c5c7 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -4027,11 +4027,13 @@
             mPendingIntent = null;
             enforceRequestCountLimit();
 
-            try {
-                mBinder.linkToDeath(this, 0);
-            } catch (RemoteException e) {
-                binderDied();
-            }
+            if (mBinder != null) {
+               try {
+                   mBinder.linkToDeath(this, 0);
+               } catch (RemoteException e) {
+                   binderDied();
+               }
+            }
         }
 
         private void enforceRequestCountLimit() {
