From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] Fix double free issue in ResTable.

diff --git a/libs/androidfw/ResourceTypes.cpp b/libs/androidfw/ResourceTypes.cpp
index 38d2a58..55078eb 100644
--- a/libs/androidfw/ResourceTypes.cpp
+++ b/libs/androidfw/ResourceTypes.cpp
@@ -3356,6 +3356,7 @@ struct ResTable::PackageGroup
                 cacheEntry.filteredConfigs.clear();
 
                 bag_set** typeBags = cacheEntry.cachedBags;
+                cacheEntry.cachedBags = NULL;
                 if (kDebugTableNoisy) {
                     printf("typeBags=%p\n", typeBags);
                 }
@@ -3368,10 +3369,10 @@ struct ResTable::PackageGroup
                     for (size_t j = 0; j < N; j++) {
                         if (typeBags[j] && typeBags[j] != (bag_set*)0xFFFFFFFF) {
                             free(typeBags[j]);
+                            typeBags[j] = NULL;
                         }
                     }
                     free(typeBags);
-                    cacheEntry.cachedBags = NULL;
                 }
             }
         }
