From 9177951ff58f0fe5dedff520975f004c2413458a Mon Sep 17 00:00:00 2001
From: Akhil Narang <akhilnarang.1999@gmail.com>
Date: Tue, 26 Jun 2018 16:55:14 +0530
Subject: [PATCH] Toast: Fix duplicated to add View of Toast

diff --git a/core/java/android/view/WindowManagerGlobal.java b/core/java/android/view/WindowManagerGlobal.java
index c7e8dee3da8..ae845796894 100644
--- a/core/java/android/view/WindowManagerGlobal.java
+++ b/core/java/android/view/WindowManagerGlobal.java
@@ -356,6 +356,7 @@ public void addView(View view, ViewGroup.LayoutParams params,
                 root.setView(view, wparams, panelParentView);
             } catch (RuntimeException e) {
                 // BadTokenException or InvalidDisplayException, clean up.
+                index = findViewLocked(view, false);
                 if (index >= 0) {
                     removeViewLocked(index, true);
                 }
