From 783c3fe87d4f0a65ea855ee10b06c50748e60c92 Mon Sep 17 00:00:00 2001
From: "Aldrin Holmes (CallMeAldy)" <aldrin.holmes20@gmail.com>
Date: Tue, 1 Sep 2015 13:02:24 +0200
Subject: [PATCH] SwipeHelper: Let's cut Max Excape Animation in Half too

diff --git a/packages/SystemUI/src/com/android/systemui/SwipeHelper.java b/packages/SystemUI/src/com/android/systemui/SwipeHelper.java
index 5c50b5c..1b27b79 100644
--- a/packages/SystemUI/src/com/android/systemui/SwipeHelper.java
+++ b/packages/SystemUI/src/com/android/systemui/SwipeHelper.java
@@ -393,7 +393,7 @@
             duration = MAX_ESCAPE_ANIMATION_DURATION;
             if (velocity != 0) {
                 duration = Math.min(duration,
-                        (int) (Math.abs(newPos - getTranslation(animView)) * 1000f / Math
+                        (int) (Math.abs(newPos - getTranslation(animView)) * 500f / Math
                                 .abs(velocity))
                 );
             } else {
