From 7079afd474c71c36a13bdb722b8bc6bd765ce7f0 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Wed, 23 Mar 2016 02:13:45 +0700
Subject: [PATCH] Don't build javacore tests

Change-Id: I1c0d0508314f9473985186a9815c19d61d50009a
---
 NativeCode.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/NativeCode.mk b/NativeCode.mk
index b21dd2d..d7f2d70 100644
--- a/NativeCode.mk
+++ b/NativeCode.mk
@@ -118,7 +118,7 @@ LOCAL_MODULE_TAGS := debug
 LOCAL_MODULE := libjavacore-unit-tests
 LOCAL_ADDITIONAL_DEPENDENCIES := $(LOCAL_PATH)/NativeCode.mk
 LOCAL_CXX_STL := libc++
-include $(BUILD_NATIVE_TEST)
+#include $(BUILD_NATIVE_TEST)
 
 # Set of benchmarks for libjavacore functions.
 include $(CLEAR_VARS)
@@ -135,7 +135,7 @@ LOCAL_CXX_STL := libc++
 LOCAL_MULTILIB := both
 LOCAL_MODULE_STEM_32 := $(LOCAL_MODULE)32
 LOCAL_MODULE_STEM_64 := $(LOCAL_MODULE)64
-include $(BUILD_NATIVE_BENCHMARK)
+#include $(BUILD_NATIVE_BENCHMARK)
 
 
 #
-- 
2.5.0

