diff --git a/src/com/android/settings/rr/input/HWSettings.java b/src/com/android/settings/rr/input/HWSettings.java
index 691c5a4..e9b6776 100644
--- a/src/com/android/settings/rr/input/HWSettings.java
+++ b/src/com/android/settings/rr/input/HWSettings.java
@@ -180,9 +180,9 @@ public class HWSettings extends ActionFragment implements
         int keysDisabled = 0;
         if (!needsNavbar) {
             mHwKeyDisable = (SwitchPreference) findPreference(HWKEY_DISABLE);
-            keysDisabled = Settings.Secure.getIntForUser(getContentResolver(),
+            keysDisabled = 0; /*Settings.Secure.getIntForUser(getContentResolver(),
                     Settings.Secure.HARDWARE_KEYS_DISABLE, 0,
-                    UserHandle.USER_CURRENT);
+                    UserHandle.USER_CURRENT);*/
             mHwKeyDisable.setChecked(keysDisabled != 0);
             mHwKeyDisable.setOnPreferenceChangeListener(this);
         } else {
@@ -299,15 +299,15 @@ public class HWSettings extends ActionFragment implements
 
     private void updateDisableNavkeysCategories() {
         final PreferenceScreen prefScreen = getPreferenceScreen();
-	    boolean enabled = Settings.Secure.getInt(getActivity().getContentResolver(),
-                Settings.Secure.HARDWARE_KEYS_DISABLE, 0) == 1;
+	    boolean enabled = false;/*Settings.Secure.getInt(getActivity().getContentResolver(),
+                Settings.Secure.HARDWARE_KEYS_DISABLE, 0) == 1;*/
         final ButtonBacklightBrightness backlight =
                 (ButtonBacklightBrightness) prefScreen.findPreference(KEY_BUTTON_BACKLIGHT);
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getContext());
         final int defaultBrightness = getContext().getResources().getInteger(com.android.internal.R.integer.config_buttonBrightnessSettingDefault);
-        int keysDisabled = Settings.Secure.getIntForUser(getContentResolver(),
+        int keysDisabled = 0; /*Settings.Secure.getIntForUser(getContentResolver(),
                     Settings.Secure.HARDWARE_KEYS_DISABLE, 0,
-                    UserHandle.USER_CURRENT);
+                    UserHandle.USER_CURRENT);*/
  	    /* Save/restore button timeouts to disable them in softkey mode */
         if (keysDisabled == 1) {
             CMSettings.Secure.putInt(getContext().getContentResolver(),
@@ -341,8 +341,9 @@ public class HWSettings extends ActionFragment implements
     public boolean onPreferenceChange(Preference preference, Object newValue) {
          if (preference == mHwKeyDisable) {
             boolean value = (Boolean) newValue;
+/*
             Settings.Secure.putInt(getContentResolver(), Settings.Secure.HARDWARE_KEYS_DISABLE,
-                    value ? 1 : 0);
+                    value ? 1 : 0); */
             setActionPreferencesEnabled(!value);
             updateDisableNavkeysCategories();
             return true;
