From e269e944609c1ef6395475b1d89b96d3ef62631d Mon Sep 17 00:00:00 2001
From: Libo Jin <liboj@codeaurora.org>
Date: Thu, 13 Jul 2017 10:57:00 +0800
Subject: [PATCH] Setting:BugFix for OOM caused by looper leak in settings

diff --git a/src/com/android/settings/dashboard/DashboardFragment.java b/src/com/android/settings/dashboard/DashboardFragment.java
index 8c3ac0d026..5bd6776461 100644
--- a/src/com/android/settings/dashboard/DashboardFragment.java
+++ b/src/com/android/settings/dashboard/DashboardFragment.java
@@ -369,4 +369,11 @@ void refreshDashboardTiles(final String TAG) {
         }
         mSummaryLoader.setListening(true);
     }
+
+    public void onDestroy(){
+        if(mSummaryLoader != null){
+           mSummaryLoader.release();
+        }
+        super.onDestroy();
+    }
 }
