From d740bcea218ee0c8060fb9ac83c2c013cb5bc09c Mon Sep 17 00:00:00 2001
From: 10190178 <cai.rui@zte.com.cn>
Date: Mon, 2 Apr 2018 10:04:01 +0800
Subject: [PATCH] Fix wifi NullPointerException crash

diff --git a/src/com/android/settings/wifi/WifiConfigController.java b/src/com/android/settings/wifi/WifiConfigController.java
index ecd4b5966e..640ee51a64 100644
--- a/src/com/android/settings/wifi/WifiConfigController.java
+++ b/src/com/android/settings/wifi/WifiConfigController.java
@@ -239,7 +239,7 @@ public WifiConfigController(WifiConfigUiBase parent, View view, AccessPoint acce
             boolean showAdvancedFields = false;
             if (mAccessPoint.isSaved()) {
                 WifiConfiguration config = mAccessPoint.getConfig();
-                if (config.getIpAssignment() == IpAssignment.STATIC) {
+                if (config != null && config.getIpAssignment() == IpAssignment.STATIC) {
                     mIpSettingsSpinner.setSelection(STATIC_IP);
                     showAdvancedFields = true;
                     // Display IP address.
