From 7d1593db6df0a4548f8ba39ff558e62ee029c9ba Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 2/7] Samsung GPS header

---
 include/hardware/gps.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/hardware/gps.h b/include/hardware/gps.h
index ff704202..6e051883 100644
--- a/include/hardware/gps.h
+++ b/include/hardware/gps.h
@@ -325,6 +325,13 @@ typedef uint8_t                         GnssConstellationType;
  */
 #define GPS_GEOFENCING_INTERFACE   "gps_geofencing"
 
+#ifdef STE_SAMSUNG_HARDWARE
+/**
+ * The GPS chipset can use Psc for AGPS.
+ */
+#define AGPS_USE_PSC
+#endif
+
 /**
  * Name of the GPS Measurements interface.
  */
@@ -387,6 +394,10 @@ typedef struct {
     float   elevation;
     /** Azimuth of SV in degrees. */
     float   azimuth;
+#ifdef STE_SAMSUNG_HARDWARE
+    /** Unknown Samsung element. */
+    int     padding;
+#endif
 } GpsSvInfo;
 
 typedef struct {
-- 
2.11.0

