From 864acad027109d526fc0ac2dd3a900b7bf0b4675 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 5/7] Fix clang

---
 modules/audio/Android.bp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules/audio/Android.bp b/modules/audio/Android.bp
index 940a5366..a8849835 100644
--- a/modules/audio/Android.bp
+++ b/modules/audio/Android.bp
@@ -27,7 +27,7 @@ cc_library_shared {
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio HAL module, identical to the default audio hal, but with
@@ -45,7 +45,7 @@ cc_library_shared {
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio policy HAL module that can be used as a skeleton for
@@ -59,7 +59,7 @@ cc_library_shared {
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio amplifier HAL module that can be used as a skeleton for
@@ -72,5 +72,5 @@ cc_library_shared {
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
-- 
2.11.0

