From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
diff --git a/modules/audio/Android.bp b/modules/audio/Android.bp
index c3cb740..6aef16a 100644
--- a/modules/audio/Android.bp
+++ b/modules/audio/Android.bp
@@ -27,7 +27,7 @@
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio HAL module, identical to the default audio hal, but with
@@ -45,7 +45,7 @@
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio policy HAL module that can be used as a skeleton for
@@ -59,7 +59,7 @@
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
 
 // The stub audio amplifier HAL module that can be used as a skeleton for
@@ -72,5 +72,5 @@
     shared_libs: [
         "liblog",
     ],
-    cflags: ["-Wall", "-Werror", "-Wno-unused-parameter"],
+    cflags: ["-Wall", "-Wno-unused-parameter"],
 }
