From a825412321e478a2b4226f4adb2da0b2acadc17e Mon Sep 17 00:00:00 2001
From: Luke Smith <luke@lukesmith.xyz>
Date: Sat, 31 Mar 2018 19:41:05 -0700
Subject: [PATCH] camera fixes

diff --git a/camera/provider/2.4/default/CameraProvider.cpp b/camera/provider/2.4/default/CameraProvider.cpp
index 19f7bdd..463d03a 100644
--- a/camera/provider/2.4/default/CameraProvider.cpp
+++ b/camera/provider/2.4/default/CameraProvider.cpp
@@ -311,6 +311,8 @@ int CameraProvider::checkCameraVersion(int id, camera_info info) {
 
 bool CameraProvider::setUpVendorTags() {
     ATRACE_CALL();
+    return true;
+#if 0
     vendor_tag_ops_t vOps = vendor_tag_ops_t();
 
     // Check if vendor operations have been implemented
@@ -363,6 +365,7 @@ bool CameraProvider::setUpVendorTags() {
         mVendorTagSections[s].tags = tagsBySection[s];
     }
     return true;
+#endif
 }
 
 // Methods from ::android::hardware::camera::provider::V2_4::ICameraProvider follow.
