From 82f9c7c222279b910db2e1a7992ecf89c861e7f9 Mon Sep 17 00:00:00 2001
From: "Ronald G. Minnich" <rminnich@gmail.com>
Date: Wed, 18 Apr 2018 04:53:18 -0700
Subject: [PATCH] wifi: set user root

diff --git a/wifi/1.1/default/android.hardware.wifi@1.0-service.rc b/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
index 770c830..3edbda6 100644
--- a/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
+++ b/wifi/1.1/default/android.hardware.wifi@1.0-service.rc
@@ -1,4 +1,4 @@
 service wifi_hal_legacy /vendor/bin/hw/android.hardware.wifi@1.0-service
     class hal
-    user wifi
+    user root
     group wifi gps
