From 4952e1709c42ccf3d9bc662883c9989f9dda9e5a Mon Sep 17 00:00:00 2001
From: Shilin Victor <radicaldreamer00001@gmail.com>
Date: Fri, 29 Mar 2019 22:51:14 +0300
Subject: [PATCH 4/4] build_W6-device-ramdisk-recovery.patch

Change-Id: I0d558c2bcebf137fbaa136d8c6e6a21999075259
---
 core/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/Makefile b/core/Makefile
index 39bf05eaa..1408bf3fa 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1190,6 +1190,7 @@ FULL_SYSTEMIMAGE_DEPS := $(INTERNAL_SYSTEMIMAGE_FILES) $(INTERNAL_USERIMAGES_DEP
 # system image.
 INSTALLED_FILES_FILE := $(PRODUCT_OUT)/installed-files.txt
 $(INSTALLED_FILES_FILE): $(FULL_SYSTEMIMAGE_DEPS)
+	$(hide) cp $(PRODUCT_OUT)/ramdisk-recovery.cpio $(PRODUCT_OUT)/install/bin/recovery.cpio
 	@echo Installed file list: $@
 	@mkdir -p $(dir $@)
 	@rm -f $@
-- 
2.11.0

