diff --git a/png.c b/png.c
index 6562cd3..e5d495b 100644
--- a/png.c
+++ b/png.c
@@ -2276,9 +2276,8 @@ png_compare_ICC_profile_with_sRGB(png_const_structrp png_ptr,
                       * problems if they are used, therefore attempt to
                       * discourage their use, skip the 'have_md5' warning below,
                       * which is made irrelevant by this error.
-                      */
                      png_chunk_report(png_ptr, "known incorrect sRGB profile",
-                        PNG_CHUNK_ERROR);
+                        PNG_CHUNK_ERROR); */
                   }
 
                   /* Warn that this being done; this isn't even an error since
@@ -2301,11 +2301,10 @@ png_compare_ICC_profile_with_sRGB(png_const_structrp png_ptr,
          /* The signature matched, but the profile had been changed in some
           * way.  This probably indicates a data error or uninformed hacking.
           * Fall through to "no match".
-          */
          png_chunk_report(png_ptr,
-             "Not recognizing known sRGB profile that has been edited", 
+             "Not recognizing known sRGB profile that has been edited",
              PNG_CHUNK_WARNING);
-         break;
+         break; */
 # endif
       }
    }
