From 60b43001c56e037e6d02811aa1eda34e3ded0a07 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <ricardo@cyngn.com>
Date: Tue, 25 Nov 2014 11:55:27 +0000
 LatinIME: Update JNI fallback lib name

diff --git a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
index 83ed5eb..0fcf7ee 100644
--- a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
+++ b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
@@ -22,5 +22,5 @@ public final class JniLibName {
     }
 
     public static final String JNI_LIB_NAME = "jni_latinime";
-    public static final String JNI_LIB_NAME2 = "jni_latinimegoogle";
+    public static final String JNI_LIB_NAME2 = "jni_unbundled_latinimegoogle";
 }
