From c736893c25541270f8ecb392276689ef9ca94852 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 14/49] binder: protected against null Parcelable

---
 libs/binder/Parcel.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 50b1bbc4a..66493512f 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -2162,6 +2162,9 @@ wp<IBinder> Parcel::readWeakBinder() const
 }
 
 status_t Parcel::readParcelable(Parcelable* parcelable) const {
+    if (parcelable == nullptr) {
+        return BAD_VALUE;
+    }
     int32_t have_parcelable = 0;
     status_t status = readInt32(&have_parcelable);
     if (status != OK) {
-- 
2.11.0

