From dfb3921ca2b485ec29dc2df994fe0e8143a80f93 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 15/43] libui: fix unresolved symbol
 GraphicBufferAllocator::alloc

---
 libs/ui/GraphicBufferAllocator.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libs/ui/GraphicBufferAllocator.cpp b/libs/ui/GraphicBufferAllocator.cpp
index 605cd2bc8..32eee0d9f 100644
--- a/libs/ui/GraphicBufferAllocator.cpp
+++ b/libs/ui/GraphicBufferAllocator.cpp
@@ -232,3 +232,6 @@ status_t GraphicBufferAllocator::free(buffer_handle_t handle)
 
 // ---------------------------------------------------------------------------
 }; // namespace android
+
+__strong_alias(_ZN7android22GraphicBufferAllocator5allocEjjiiPPK13native_handlePi,
+        _ZN7android22GraphicBufferAllocator5allocEjjijPPK13native_handlePj);
-- 
2.11.0

