From 0f221c9ccdebc33ba5405426321618e2bf20fc89 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 26 Dec 2016 19:45:41 +0700
Subject: [PATCH 18/26] frameworks-native_WW02-fix-use-min-realtime.patch

---
 services/surfaceflinger/surfaceflinger.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/surfaceflinger/surfaceflinger.rc b/services/surfaceflinger/surfaceflinger.rc
index 435aa0c..6b08be2 100644
--- a/services/surfaceflinger/surfaceflinger.rc
+++ b/services/surfaceflinger/surfaceflinger.rc
@@ -2,5 +2,7 @@ service surfaceflinger /system/bin/surfaceflinger
     class core
     user system
     group graphics drmrpc readproc
+    # allow the minimum realtime scheduler priority: RLIMIT_RTPRIO(14)=1
+    rlimit 14 1 1
     onrestart restart zygote
     writepid /dev/stune/foreground/tasks
-- 
2.9.3

