From 19bb085028e3e0001296a3b0d25d6094c71dc050 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 31/46] libbinder: fix unresolved symbol WriteString16

---
 libs/binder/Parcel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 81777f4d82..3c77616667 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -2855,3 +2855,5 @@ void Parcel::Blob::clear() {
 }
 
 }; // namespace android
+
+__strong_alias(_ZN7android6Parcel13writeString16EPKtj,_ZN7android6Parcel13writeString16EPKDsj);
-- 
2.17.0

