From 74ebdf246c562175b40a792e2b5523a9dc5d5494 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 39/43] binder: protected against null Parcelable

---
 libs/binder/Parcel.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index d368d712e..8b9e4922d 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -2088,6 +2088,9 @@ wp<IBinder> Parcel::readWeakBinder() const
 }
 
 status_t Parcel::readParcelable(Parcelable* parcelable) const {
+    if (parcelable == nullptr) {
+        return BAD_VALUE;
+    }
     int32_t have_parcelable = 0;
     status_t status = readInt32(&have_parcelable);
     if (status != OK) {
-- 
2.11.0

