From 6b123f4c34c88e69ae5de98eefcd8163189dbcbd Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 10/17] libril: enable mobile data fix for STE hardware

---
 libril/ril_service.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libril/ril_service.cpp b/libril/ril_service.cpp
index 10f11dd..66067fa 100644
--- a/libril/ril_service.cpp
+++ b/libril/ril_service.cpp
@@ -7418,7 +7418,7 @@ void convertRilDataCallToHal(RIL_Data_Call_Response_v6 *dcResponse,
     dcResult.ifname = convertCharPtrToHidlString(dcResponse->ifname);
     dcResult.addresses = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.dnses = convertCharPtrToHidlString(dcResponse->dnses);
-    dcResult.gateways = convertCharPtrToHidlString(dcResponse->gateways);
+    dcResult.gateways = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.pcscf = hidl_string();
     dcResult.mtu = 0;
 }
@@ -7433,7 +7433,7 @@ void convertRilDataCallToHal(RIL_Data_Call_Response_v9 *dcResponse,
     dcResult.ifname = convertCharPtrToHidlString(dcResponse->ifname);
     dcResult.addresses = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.dnses = convertCharPtrToHidlString(dcResponse->dnses);
-    dcResult.gateways = convertCharPtrToHidlString(dcResponse->gateways);
+    dcResult.gateways = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.pcscf = convertCharPtrToHidlString(dcResponse->pcscf);
     dcResult.mtu = 0;
 }
@@ -7448,7 +7448,7 @@ void convertRilDataCallToHal(RIL_Data_Call_Response_v11 *dcResponse,
     dcResult.ifname = convertCharPtrToHidlString(dcResponse->ifname);
     dcResult.addresses = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.dnses = convertCharPtrToHidlString(dcResponse->dnses);
-    dcResult.gateways = convertCharPtrToHidlString(dcResponse->gateways);
+    dcResult.gateways = convertCharPtrToHidlString(dcResponse->addresses);
     dcResult.pcscf = convertCharPtrToHidlString(dcResponse->pcscf);
     dcResult.mtu = dcResponse->mtu;
 }
-- 
2.11.0

